https://wikidocs.net/book/922

초보자를 위한 파이썬 300제

# <01. 파이썬 시작하기>

# 001 print 기초
화면에 Hello World 문자열을 출력하세요.

In [1]:
print("Hello World")

Hello World


# 002 print 기초
화면에 Mary's cosmetics을 출력하세요. (중간에 '가 있음에 주의하세요)

In [5]:
print("Mary's cosmetics")

Mary's cosmetics


# 003 print 기초
화면에 아래 문장을 출력하세요. (중간에 "가 있음에 주의하세요.)

In [6]:
print('신씨가 소리질렀다. "도둑이야".')

신씨가 소리질렀다. "도둑이야".


# 004 print 기초
화면에 "C:\Windows"를 출력하세요.

In [7]:
print('"C:\Windows"')

"C:\Windows"


# 005 print 탭과 줄바꿈
다음 코드를 실행해보고 \t와 \n의 역할을 설명해보세요.

print("안녕하세요.\n만나서\t\t반갑습니다.")

In [8]:
print("안녕하세요.\n만나서\t\t반갑습니다.")

안녕하세요.
만나서		반갑습니다.


\t 는 tab을 출력

\n 은 줄바꿈

# 006 print 여러 데이터 출력
print 함수에 두 개의 단어를 입력한 예제입니다. 아래 코드의 출력 결과를 예상해봅시다.

print ("오늘은", "일요일")

오늘은 일요일

이라고 출력 될 것이다.

In [9]:
print ("오늘은", "일요일")

오늘은 일요일


# 007 print 기초
print() 함수를 사용하여 다음과 같이 출력하세요.

naver;kakao;sk;samsung

In [14]:
print("naver;kakao;sk;samsung")

naver;kakao;sk;samsung


In [15]:
print('naver','kakao','sk','samsung',sep=';')

naver;kakao;sk;samsung


# 008 print 기초
print() 함수를 사용하여 다음과 같이 출력하세요.

naver/kakao/sk/samsung

In [13]:
print('naver/kakao/sk/samsung')

naver/kakao/sk/samsung


In [16]:
print('naver','kakao','sk','samsung',sep='/')

naver/kakao/sk/samsung


# 009 print 줄바꿈
다음 코드를 수정하여 줄바꿈이 없이 출력하세요. (힌트: end='') print 함수는 두 번 사용합니다. 세미콜론 (;)은 한줄에 여러 개의 명령을 작성하기 위해 사용합니다.

print("first");print("second")

In [18]:
print("first",end=' ')
print("second")

first second


# 010 연산 결과 출력
5/3의 결과를 화면에 출력하세요.

In [19]:
print(5/3)

1.6666666666666667


# <02. 파이썬 변수>

# 011 변수 사용하기
삼성전자라는 변수로 50,000원을 바인딩해보세요. 삼성전자 주식 10주를 보유하고 있을 때 총 평가금액을 출력하세요.

In [22]:
삼성전자 = 50000
print(삼성전자*10,'원',sep='')

500000원


# 012 변수 사용하기
다음 표는 삼성전자의 일부 투자정보입니다. 변수를 사용해서 시가총액, 현재가, PER 등을 바인딩해보세요.

바인딩 : 프로그램에 사용된 구성 요소의 실제 값 또는 프로퍼티를 결정짓는 행위

In [1]:
시가총액 = 2980000000000
현재가 = 50000
PER = 15.79
print(시가총액,type(시가총액))
print(현재가,type(현재가))
print(PER,type(PER))

2980000000000 <class 'int'>
50000 <class 'int'>
15.79 <class 'float'>


# 013 문자열 출력
변수 s와 t에는 각각 문자열이 바인딩 되어있습니다.

In [2]:
s="hello"
t="python"

두 변수를 이용하여 아래와 같이 출력해보세요.

실행 예:
hello! python

In [4]:
s+='!'
print(s,t)

hello! python


# 014 파이썬을 이용한 값 계산
아래 코드의 실행 결과를 예상해보세요.

>> 2 + 2 * 3 

답 : 8

In [5]:
2 + 2*3

8

# 015 type 함수

type() 함수는 데이터 타입을 판별합니다. 변수 a에는 128 숫자가 바인딩돼 있어 type 함수가 int (정수)형임을 알려줍니다.

>> a = 128

>> print (type(a))

<class 'int'>

아래 변수에 바인딩된 값의 타입을 판별해보세요.

>> a = "132"


In [7]:
a="132"
print(type(a))

<class 'str'>


# 016 문자열을 정수로 변환
문자열 '720'를 정수형으로 변환해보세요.

>> num_str = "720"

In [9]:
num_str = "720"
num_str2 = int(num_str)
print(type(num_str2))

<class 'int'>


# 017 정수를 문자열 100으로 변환
정수 100을 문자열 '100'으로 변환해보세요.

num = 100

In [10]:
num = 100
num2 = str(num)
print(type(num2))

<class 'str'>


# 018 문자열을 실수로변환
문자열 "15.79"를 실수(float) 타입으로 변환해보세요.

In [14]:
x = "15.79"
ans = float(x)
print(type(ans))

<class 'float'>


# 019 문자열을 정수로 변환
year라는 변수가 문자열 타입의 연도를 바인딩하고 있습니다. 이를 정수로 변환한 후 최근 3년의 연도를 화면에 출력해보세요.

year = "2020"

In [15]:
year = "2020"
ans = int(year)
for i in range(3,0,-1):
    print(ans-i)

2017
2018
2019


# 020 파이썬 계산
에이컨이 월 48,584원에 무이자 36개월의 조건으로 홈쇼핑에서 판매되고 있습니다. 총 금액은 계산한 후 이를 화면에 출력해보세요. (변수사용하기)



In [16]:
air = 48584
month = 36
ans = air*month
print(ans)

1749024


# <파이썬 문자열>

# 021 문자열 인덱싱
letters가 바인딩하는 문자열에서 첫번째와 세번째 문자를 출력하세요.

>> letters = 'python'

In [17]:
letters = 'python'
print(letters[0],letters[2])

p t


# 022 문자열 슬라이싱
자동차 번호가 다음과 같을 때 뒤에 4자리만 출력하세요.

>> license_plate = "24가 2210"

In [18]:
license_plate = "24가 2210"
print(license_plate[4:])

2210


# 023 문자열 인덱싱
아래의 문자열에서 '홀' 만 출력하세요.

>> string = "홀짝홀짝홀짝"

In [20]:
string = "홀짝홀짝홀짝"
for i in range(len(string)):
    if string[i]=='홀':
        print('홀',end='')

홀홀홀

# 024 문자열 슬라이싱
문자열을 거꾸로 뒤집어 출력하세요.

>> string = "PYTHON"

In [22]:
string='PYTHON'
print(string[::-1])

NOHTYP


# 025 문자열 치환
아래의 전화번호에서 하이푼 ('-')을 제거하고 출력하세요.

>> phone_number = "010-1111-2222"

In [24]:
phone_number = "010-1111-2222"
print(phone_number.replace("-"," "))

010 1111 2222


# 026 문자열 다루기
25번 문제의 전화번호를 아래와 같이 모두 붙여 출력하세요.



In [27]:
print(phone_number.replace("-",""))

01011112222


# 027 문자열 다루기

url 에 저장된 웹 페이지 주소에서 도메인을 출력하세요.

>> url = "http://sharebook.kr"

In [30]:
url = "http://sharebook.kr"
print(url[-2:])

kr


# 028 문자열은 immutable
아래 코드의 실행 결과를 예상해보세요.

>> lang = 'python'

>> lang[0] = 'P'

>> print(lang)

에러발생

문자열은 저 형식으로 수정 불가

# 029 replace 메서드
아래 문자열에서 소문자 'a'를 대문자 'A'로 변경하세요.

>> string = 'abcdfe2a354a32a'

In [32]:
string = 'abcdfe2a354a32a'
string2 = string.replace('a','A')
print(string2)

Abcdfe2A354A32A


# 030 replace 메서드
아래 코드의 실행 결과를 예상해보세요.

>> string = 'abcd'

>> string.replace('b', 'B')

>> print(string)

답 : abcd 그대로 출력

In [34]:
string = 'abcd'
string.replace('b', 'B')
print(string)

abcd


# 031 문자열 합치기
아래 코드의 실행 결과를 예상해보세요.

>> a = "3"

>> b = "4"

>> print(a + b)

답 : 34출력

In [35]:
a = "3"
b = "4"
print(a + b)

34


# 032 문자열 곱하기
아래 코드의 실행 결과를 예상해보세요.

>> print("Hi" * 3)

답 : HiHiHi 출력

In [37]:
print("Hi" * 3)

HiHiHi


# 033 문자열 곱하기

화면에 '-'를 80개 출력하세요.

In [38]:
print('-'*80)

--------------------------------------------------------------------------------


# 034 문자열 곱하기
변수에 다음과 같은 문자열이 바인딩되어 있습니다.

>>> t1 = 'python'

>>> t2 = 'java'

변수에 문자열 더하기와 문자열 곱하기를 사용해서 아래와 같이 출력해보세요.

실행 예:
python java python java python java python java

In [46]:
t1 = 'python'
t2 = 'java'
ans = t1+' '+t2+' '
print((ans)*4)

python java python java python java python java 


# 035 문자열 출력
변수에 다음과 같이 문자열과 정수가 바인딩되어 있을 때 % formatting을 사용해서 다음과 같이 출력해보세요.

name1 = "김민수" 

age1 = 10

name2 = "이철희"

age2 = 13

이름: 김민수 나이: 10

이름: 이철희 나이: 13

In [47]:
name1 = "김민수" 
age1 = 10
name2 = "이철희"
age2 = 13
print("이름 : %s 나이: %d" %(name1,age1))
print("이름 : %s 나이: %d" %(name2,age2))

이름 : 김민수 나이: 10
이름 : 이철희 나이: 13


# 036 문자열 출력
문자열의 format( ) 메서드를 사용해서 035번 문제를 다시 풀어보세요.

In [48]:
print("이름: {} 나이: {}".format(name1, age1))
print("이름: {} 나이: {}".format(name2, age2))

이름: 김민수 나이: 10
이름: 이철희 나이: 13


# 037 문자열 출력
파이썬 3.6부터 지원하는 f-string을 사용해서 035번 문제를 다시 풀어보세요.

In [49]:
print(f"이름: {name1} 나이: {age1}")
print(f"이름: {name2} 나이: {age2}")

이름: 김민수 나이: 10
이름: 이철희 나이: 13


# 038 컴마 제거하기
삼성전자의 상장주식수가 다음과 같습니다. 컴마를 제거한 후 이를 정수 타입으로 변환해보세요.

상장주식수 = "5,969,782,550"

In [51]:
상장주식수 = "5,969,782,550"
ans = 상장주식수.replace(",","")
ans = int(ans)
print(ans,type(ans))

5969782550 <class 'int'>


# 039 문자열 슬라이싱
다음과 같은 문자열에서 '2020/03'만 출력하세요.

분기 = "2020/03(E) (IFRS연결)"

In [53]:
분기 = "2020/03(E) (IFRS연결)"
print(분기[:7])

2020/03


# 040 strip 메서드
문자열의 좌우의 공백이 있을 때 이를 제거해보세요.

data = "   삼성전자    "

In [54]:
data = "   삼성전자    "
ans = data.strip()
print(ans)

삼성전자


# 041 upper 메서드
다음과 같은 문자열이 있을 때 이를 대문자 BTC_KRW로 변경하세요.

ticker = "btc_krw"

In [56]:
ticker = "btc_krw"
ans = ticker.upper()
print(ans)

BTC_KRW


# 042 lower 메서드
다음과 같은 문자열이 있을 때 이를 소문자 btc_krw로 변경하세요.

ticker = "BTC_KRW"

In [57]:
ticker = "BTC_KRW"
ans = ticker.lower()
print(ans)

btc_krw


# 043 capitalize 메서드
문자열 'hello'가 있을 때 이를 'Hello'로 변경해보세요.

In [58]:
a = 'hello'
ans = a.capitalize()
print(ans)

Hello


# 044 endswith 메서드
파일 이름이 문자열로 저장되어 있을 때 endswith 메서드를 사용해서 파일 이름이 'xlsx'로 끝나는지 확인해보세요.

file_name = "보고서.xlsx"

In [62]:
file_name = "보고서.xlsx"
ans = file_name.endswith("xlsx")
print(ans)

True


# 045 endswith 메서드
파일 이름이 문자열로 저장되어 있을 때 endswith 메서드를 사용해서 파일 이름이 'xlsx' 또는 'xls'로 끝나는지 확인해보세요.

file_name = "보고서.xlsx"

In [64]:
file_name = "보고서.xlsx"
ans = file_name.endswith(("xlsx", "xls"))
print(ans)

True


# 046 startswith 메서드
파일 이름이 문자열로 저장되어 있을 때 startswith 메서드를 사용해서 파일 이름이 '2020'로 시작하는지 확인해보세요.

file_name = "2020_보고서.xlsx"

In [65]:
file_name = "2020_보고서.xlsx"
ans = file_name.startswith("2020")
print(ans)

True


# 047 split 메서드
다음과 같은 문자열이 있을 때 공백을 기준으로 문자열을 나눠보세요.

a = "hello world"

In [67]:
a="hello world"
ans = a.split()
print(ans)

['hello', 'world']


In [68]:
type(ans)

list

# 048 split 메서드
다음과 같이 문자열이 있을 때 btc와 krw로 나눠보세요.

ticker = "btc_krw"

In [69]:
ticker = "btc_krw"
ans = ticker.split('_')
print(ans)

['btc', 'krw']


# 049 split 메서드
다음과 같이 날짜를 표현하는 문자열이 있을 때 연도, 월, 일로 나눠보세요.

date = "2020-05-01"

In [72]:
date = "2020-05-01"
ans = date.split('-')
print(ans)

['2020', '05', '01']


# 050 rstrip 메서드

문자열의 오른쪽에 공백이 있을 때 이를 제거해보세요.

data = "039490     "

In [73]:
data = "039490     "
ans=data.rstrip()
print(ans)

039490


# <04. 파이썬 리스트>

# 051 리스트 생성
2016년 11월 영화 예매 순위 기준 top3는 다음과 같습니다. 영화 제목을 movie_rank 이름의 리스트에 저장해보세요. (순위 정보는 저장하지 않습니다.)

In [74]:
movie_rank=['닥터스트레인지','스플릿','럭키']
print(movie_rank)

['닥터스트레인지', '스플릿', '럭키']


# 052 리스트에 원소추가
051의 movie_rank 리스트에 "배트맨"을 추가하라.

In [75]:
movie_rank.append('배트맨')
print(movie_rank)

['닥터스트레인지', '스플릿', '럭키', '배트맨']


# 053
movie_rank 리스트에는 아래와 같이 네 개의 영화 제목이 바인딩되어 있다. "슈퍼맨"을 "닥터 스트레인지"와 "스플릿" 사이에 추가하라.

movie_rank = ['닥터 스트레인지', '스플릿', '럭키', '배트맨']

In [76]:
movie_rank.insert(1, "슈퍼맨")
print(movie_rank)

['닥터스트레인지', '슈퍼맨', '스플릿', '럭키', '배트맨']


# 054
movie_rank 리스트에서 '럭키'를 삭제하라.

movie_rank = ['닥터 스트레인지', '슈퍼맨', '스플릿', '럭키', '배트맨']

In [78]:
del movie_rank[3]
print(movie_rank)

['닥터스트레인지', '슈퍼맨', '스플릿', '배트맨']


# 055
movie_rank 리스트에서 '스플릿' 과 '배트맨'을 를 삭제하라.

movie_rank = ['닥터 스트레인지', '슈퍼맨', '스플릿', '배트맨']

In [79]:
del movie_rank[1]
del movie_rank[1]
print(movie_rank)

['닥터스트레인지', '배트맨']


# 056
lang1과 lang2 리스트가 있을 때 lang1과 lang2의 원소를 모두 갖고 있는 langs 리스트를 만들어라.

>> lang1 = ["C", "C++", "JAVA"]

>> lang2 = ["Python", "Go", "C#"]

In [80]:
lang1 = ["C", "C++", "JAVA"]
lang2 = ["Python", "Go", "C#"]
ans = lang1+lang2
print(ans)

['C', 'C++', 'JAVA', 'Python', 'Go', 'C#']


# 057
다음 리스트에서 최댓값과 최솟값을 출력하라. (힌트: min(), max() 함수 사용)

nums = [1, 2, 3, 4, 5, 6, 7]

In [82]:
nums = [1, 2, 3, 4, 5, 6, 7]
print(min(nums),max(nums))

1 7


# 058
다음 리스트의 합을 출력하라.

nums = [1, 2, 3, 4, 5]

In [84]:
nums = [1, 2, 3, 4, 5]
ans = sum(nums)
print(ans)

15


# 059
다음 리스트에 저장된 데이터의 개수를 화면에 구하하라.

cook = ["피자", "김밥", "만두", "양념치킨", "족발", "피자", "김치만두", "쫄면", "소시지", "라면", "팥빙수", "김치전"]

In [85]:
cook = ["피자", "김밥", "만두", "양념치킨", "족발", "피자", "김치만두", "쫄면", "소시지", "라면", "팥빙수", "김치전"]
ans = len(cook)
print(ans)

12


# 060
다음 리스트의 평균을 출력하라.

nums = [1, 2, 3, 4, 5]

In [88]:
nums=[1,2,3,4,5]
ans = sum(nums)/len(nums)
print(ans)

3.0


# 061
price 변수에는 날짜와 종가 정보가 저장돼 있다. 날짜 정보를 제외하고 가격 정보만을 출력하라. (힌트 : 슬라이싱)

price = ['20180728', 100, 130, 140, 150, 160, 170]

In [89]:
price = ['20180728', 100, 130, 140, 150, 160, 170]
print(price[1:])

[100, 130, 140, 150, 160, 170]


# 062
슬라이싱을 사용해서 홀수만 출력하라.

nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

In [90]:
nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
print(nums[::2])

[1, 3, 5, 7, 9]


# 063
슬라이싱을 사용해서 짝수만 출력하라.

nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

In [91]:
nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
print(nums[1::2])

[2, 4, 6, 8, 10]


# 064
슬라이싱을 사용해서 리스트의 숫자를 역 방향으로 출력하라.

nums = [1, 2, 3, 4, 5]

In [92]:
nums = [1, 2, 3, 4, 5]
print(nums[::-1])

[5, 4, 3, 2, 1]


# 065
interest 리스트에는 아래의 데이터가 바인딩되어 있다.

interest = ['삼성전자', 'LG전자', 'Naver']

interest 리스트를 사용하여 아래와 같이 화면에 출력하라.

출력 예시:
삼성전자 Naver

In [94]:
interest = ['삼성전자', 'LG전자', 'Naver']
print(interest[0],interest[-1])

삼성전자 Naver


# 066
interest 리스트에는 아래의 데이터가 바인딩되어 있다.

interest = ['삼성전자', 'LG전자', 'Naver', 'SK하이닉스', '미래에셋대우']
interest 리스트를 사용하여 아래와 같이 화면에 출력하라.

출력 예시:
삼성전자 LG전자 Naver SK하이닉스 미래에셋대우

In [97]:
interest = ['삼성전자', 'LG전자', 'Naver', 'SK하이닉스', '미래에셋대우']
print(" ".join(interest))

삼성전자 LG전자 Naver SK하이닉스 미래에셋대우


# 067
interest 리스트에는 아래의 데이터가 바인딩되어 있다.

interest = ['삼성전자', 'LG전자', 'Naver', 'SK하이닉스', '미래에셋대우']

interest 리스트를 사용하여 아래와 같이 화면에 출력하라.

출력 예시:
삼성전자/LG전자/Naver/SK하이닉스/미래에셋대우

In [99]:
interest = ['삼성전자', 'LG전자', 'Naver', 'SK하이닉스', '미래에셋대우']
print("/".join(interest))

삼성전자/LG전자/Naver/SK하이닉스/미래에셋대우


# 068
interest 리스트에는 아래의 데이터가 바인딩되어 있다.

interest = ['삼성전자', 'LG전자', 'Naver', 'SK하이닉스', '미래에셋대우']

join() 메서드를 사용해서 interest 리스트를 아래와 같이 화면에 출력하라.

출력 예시:

삼성전자

LG전자

Naver

SK하이닉스

미래에셋대우


In [100]:
interest = ['삼성전자', 'LG전자', 'Naver', 'SK하이닉스', '미래에셋대우']
print("\n".join(interest))

삼성전자
LG전자
Naver
SK하이닉스
미래에셋대우


# 069
회사 이름이 슬래시 ('/')로 구분되어 하나의 문자열로 저장되어 있다.

string = "삼성전자/LG전자/Naver"

이를 interest 이름의 리스트로 분리 저장하라.

실행 예시

>> print(interest)

['삼성전자', 'LG전자', 'Naver']

In [102]:
string = "삼성전자/LG전자/Naver"
interest = string.split('/')
print(interest)

['삼성전자', 'LG전자', 'Naver']


# 070 리스트 정렬
리스트에 있는 값을 오름차순으로 정렬하세요.

data = [2, 4, 3, 1, 5, 10, 9]

In [103]:
data = [2, 4, 3, 1, 5, 10, 9]
data.sort()
print(data)

[1, 2, 3, 4, 5, 9, 10]


# <05. 파이썬 튜플>

# 071
my_variable 이름의 비어있는 튜플을 만들라.

In [104]:
my_variable=()

# 072
2016년 11월 영화 예매 순위 기준 top3는 다음과 같다. 영화 제목을 movie_rank 이름의 튜플에 저장하라. (순위 정보는 저장하지 않는다.)

In [106]:
movie_rank=('닥터스트레인지','스플릿','럭키')
print(movie_rank)

('닥터스트레인지', '스플릿', '럭키')


# 073
숫자 1 이 저장된 튜플을 생성하라.

In [109]:
ans = (1,)
print(ans)

(1,)


# 074
다음 코드를 실행해보고 오류가 발생하는 원인을 설명하라.

>> t = (1, 2, 3)

>> t[0] = 'a'

Traceback (most recent call last):

File "<pyshell#46>", line 1, in <module>

    t[0] = 'a'

    TypeError: 'tuple' object does not support item assignment

In [111]:
t = (1, 2, 3)
t[0] = 'a'

TypeError: 'tuple' object does not support item assignment

답 : 튜플은 원소 값 변경 불가

# 075
아래와 같이 t에는 1, 2, 3, 4 데이터가 바인딩되어 있다. t가 바인딩하는 데이터 타입은 무엇인가?

t = 1, 2, 3, 4


답 : tuple 

In [112]:
t = 1,2,3,4
type(t)

tuple

# 076
변수 t에는 아래와 같은 값이 저장되어 있다. 변수 t가 ('A', 'b', 'c') 튜플을 가리키도록 수정 하라.

t = ('a', 'b', 'c')

In [113]:
t = ('a', 'b', 'c')
t = ('A', 'b', 'c')
print(t)

('A', 'b', 'c')


# 077
다음 튜플을 리스트로 변환하라.

interest = ('삼성전자', 'LG전자', 'SK Hynix')

In [115]:
interest = ('삼성전자', 'LG전자', 'SK Hynix')
ans=list(interest)
print(ans)

['삼성전자', 'LG전자', 'SK Hynix']


# 078
다음 리스트를 튜플로 변경하라.

interest = ['삼성전자', 'LG전자', 'SK Hynix']

In [116]:
interest = ['삼성전자', 'LG전자', 'SK Hynix']
ans = tuple(interest)
print(ans)

('삼성전자', 'LG전자', 'SK Hynix')


# 079
다음 코드의 실행 결과를 예상하라.

temp = ('apple', 'banana', 'cake')

a, b, c = temp

print(a, b, c)

답 : apple banana cake 출력

In [117]:
temp = ('apple', 'banana', 'cake')
a, b, c = temp
print(a, b, c)


apple banana cake


# 080 range 함수
1 부터 99까지의 정수 중 짝수만 저장된 튜플을 생성하라.

(2, 4, 6, 8 ... 98)

In [120]:
ans = tuple(range(2,100,2))
print(ans)

(2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48, 50, 52, 54, 56, 58, 60, 62, 64, 66, 68, 70, 72, 74, 76, 78, 80, 82, 84, 86, 88, 90, 92, 94, 96, 98)


# <06. 파이썬 딕셔너리>

# 081 별 표현식
기본적으로 데이터 언패킹은 좌변의 변수와 우변 데이터 개수가 같아야 합니다. 하지만 star expression을 사용하면 변수의 개수가 달라도 데이터 언패킹을 할 수 있습니다. 튜플에 저장된 데이터 중에서 앞에 있는 두 개의 데이터만 필요할 경우 나머지 데이터의 언패킹 코드를 작성할 필요가 없습니다.

a, b, *c = (0, 1, 2, 3, 4, 5)

>a

0

>b

1

>c

[2, 3, 4, 5]

다음과 같이 10개의 값이 저장된 scores 리스트가 있을 때, start expression을 사용하여 좌측 8개의 값을 valid_score 변수에 바인딩하여라.

scores = [8.8, 8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8, 9.4]

In [122]:
scores = [8.8, 8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8, 9.4]
*valid_score, a, b= scores
print(valid_score)

[8.8, 8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5]


# 082
다음과 같이 10개의 값이 저장된 scores 리스트가 있을 때, start expression을 사용하여 우측 8개의 값을 valid_score 변수에 바인딩하여라.

scores = [8.8, 8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8, 9.4]

In [124]:
scores = [8.8, 8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8, 9.4]
a,b,*valid_score = scores
print(valid_score)

[8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8, 9.4]


# 083
다음과 같이 10개의 값이 저장된 scores 리스트가 있을 때, start expression을 사용하여 가운데 있는 8개의 값을 valid_score 변수에 바인딩하여라.

scores = [8.8, 8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8, 9.4]

In [125]:
scores = [8.8, 8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8, 9.4]
a,*valid_score,b=scores
print(valid_score)

[8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8]


# 084 비어있는 딕셔너리
temp 이름의 비어있는 딕셔너리를 만들어라

In [126]:
temp = {}
print(temp,type(temp))

{} <class 'dict'>


# 085
다음 아이스크림 이름과 희망 가격을 딕셔너리로 구성하라.

In [127]:
ans = {'메로나':1000, '폴라포':1200, '빵빠레':1800}
print(ans)

{'메로나': 1000, '폴라포': 1200, '빵빠레': 1800}


# 086
085 번의 딕셔너리에 아래 아이스크림 가격정보를 추가하라.

In [128]:
ans["죠스바"]=1200
ans["월드콘"]=1500
print(ans)

{'메로나': 1000, '폴라포': 1200, '빵빠레': 1800, '죠스바': 1200, '월드콘': 1500}


# 087
다음 딕셔너리를 사용하여 메로나 가격을 출력하라.

In [129]:
print(f'메로나 가격: {ans["메로나"]}')

메로나 가격: 1000


# 088
다음 딕셔너리에서 메로나의 가격을 1300으로 수정하라.

In [130]:
ans['메로나']=1300
print(ans)

{'메로나': 1300, '폴라포': 1200, '빵빠레': 1800, '죠스바': 1200, '월드콘': 1500}


# 089
다음 딕셔너리에서 메로나를 삭제하라.

In [131]:
ans = {'메로나': 1000, '폴라포': 1200, '빵빠레': 1800, '죠스바': 1200, '월드콘': 1500}

In [133]:
del ans["메로나"]
print(ans)

{'폴라포': 1200, '빵빠레': 1800, '죠스바': 1200, '월드콘': 1500}


# 090
다음 코드에서 에러가 발생한 원인을 설명하라.

>> icecream = {'폴라포': 1200, '빵빠레': 1800, '월드콘': 1500, '메로나': 1000}

>> icecream['누가바']

Traceback (most recent call last):

File "<pyshell#69>", line 1, in <module>

    icecream['누가바']

KeyError: '누가바'

답 : icecream 딕셔너리에 누가바 라는 키가 없음

# 091 딕셔너리 생성
아래의 표에서, 아이스크림 이름을 키값으로, (가격, 재고) 리스트를 딕셔너리의 값으로 저장하라. 딕셔너리의 이름은 inventory로 한다.

In [134]:
inventory={'메로나':[300,20],'비비빅':[400,3],'죠스바':[250,100]}
print(inventory)

{'메로나': [300, 20], '비비빅': [400, 3], '죠스바': [250, 100]}


# 092 딕셔너리 인덱싱
inventory 딕셔너리에서 메로나의 가격을 화면에 출력하라.

inventory = {"메로나": [300, 20],
              "비비빅": [400, 3],
              "죠스바": [250, 100]}

In [135]:
print(f'{inventory["메로나"][0]} 원')

300 원


# 093 딕셔너리 인덱싱
inventory 딕셔너리에서 메로나의 재고를 화면에 출력하라.

In [136]:
print(f'{inventory["메로나"][1]} 개')

20 개


# 094 딕셔너리 추가
inventory 딕셔너리에 아래 데이터를 추가하라.

In [1]:
inventory = {"메로나": [300, 20],"비비빅": [400, 3],"죠스바": [250, 100]}

In [2]:
inventory["월드콘"]=[500,7]
print(inventory)

{'메로나': [300, 20], '비비빅': [400, 3], '죠스바': [250, 100], '월드콘': [500, 7]}


# 095 딕셔너리 keys() 메서드
다음의 딕셔너리로부터 key 값으로만 구성된 리스트를 생성하라.

icecream = {'탱크보이': 1200, '폴라포': 1200, '빵빠레': 1800, '월드콘': 1500, '메로나': 1000}

In [3]:
icecream = {'탱크보이': 1200, '폴라포': 1200, 
            '빵빠레': 1800, '월드콘': 1500, '메로나': 1000}
ans = list(icecream.keys())
print(ans)

['탱크보이', '폴라포', '빵빠레', '월드콘', '메로나']


# 096 딕셔너리 values() 메서드

다음의 딕셔너리에서 values 값으로만 구성된 리스트를 생성하라.

icecream = {'탱크보이': 1200, '폴라포': 1200, '빵빠레': 1800, '월드콘': 1500, '메로나': 1000}

In [4]:
icecream = {'탱크보이': 1200, '폴라포': 1200, '빵빠레': 1800, 
            '월드콘': 1500, '메로나': 1000}
ans = list(icecream.values())
print(ans)

[1200, 1200, 1800, 1500, 1000]


# 097 딕셔너리 values() 메서드

In [5]:
icecream = {'탱크보이': 1200, '폴라포': 1200, '빵빠레': 1800, 
            '월드콘': 1500, '메로나': 1000}
ans = list(icecream.values())
print(sum(ans))

6700


# 098 딕셔너리 update 메서드

아래의 new_product 딕셔너리를 다음 icecream 딕셔너리에 추가하라.

icecream = {'탱크보이': 1200, '폴라포': 1200, '빵빠레': 1800, '월드콘': 1500, '메로나': 1000}
new_product = {'팥빙수':2700, '아맛나':1000}

In [10]:
icecream = {'탱크보이': 1200, '폴라포': 1200, '빵빠레': 1800, 
            '월드콘': 1500, '메로나': 1000}
new_product = {'팥빙수':2700, '아맛나':1000}
icecream.update(new_product)
print(icecream)

{'탱크보이': 1200, '폴라포': 1200, '빵빠레': 1800, '월드콘': 1500, '메로나': 1000, '팥빙수': 2700, '아맛나': 1000}


# 099 zip과 dict

아래 두 개의 튜플을 하나의 딕셔너리로 변환하라. keys를 키로, vals를 값으로 result 이름의 딕셔너리로 저장한다.

keys = ("apple", "pear", "peach")
vals = (300, 250, 400)

In [12]:
keys = ("apple", "pear", "peach")
vals = (300, 250, 400)
result = dict(zip(keys,vals))
print(result)

{'apple': 300, 'pear': 250, 'peach': 400}


# 100 zip과 dict

date와 close_price 두 개의 리스트를 close_table 이름의 딕셔너리로 생성하라.

date = ['09/05', '09/06', '09/07', '09/08', '09/09']
close_price = [10500, 10300, 10100, 10800, 11000]

In [14]:
date = ['09/05', '09/06', '09/07', '09/08', '09/09']
close_price = [10500, 10300, 10100, 10800, 11000]
close_table = dict(zip(date,close_price))
print(close_table)

{'09/05': 10500, '09/06': 10300, '09/07': 10100, '09/08': 10800, '09/09': 11000}


# <07. 파이썬 분기문>

# 101
파이썬에서 True 혹은 False를 갖는 데이터 타입은 무엇인가?

답 : bool

# 102
아래 코드의 출력 결과를 예상하라

print(3 == 5)

답 : False

In [15]:
print(3 == 5)

False


# 103
아래 코드의 출력 결과를 예상하라

print(3 < 5)

답 : True

In [16]:
print(3<5)

True


# 104
아래 코드의 결과를 예상하라.

x = 4

print(1 < x < 5)

답 : True

In [17]:
x = 4
print(1 < x < 5)

True


# 105
아래 코드의 결과를 예상하라.

print ((3 == 3) and (4 != 3))

답 : True

In [18]:
print ((3 == 3) and (4 != 3))

True


# 106
아래 코드에서 에러가 발생하는 원인에 대해 설명하라.

print(3 => 4)

답 : => 이 아니라 >=이라 입력하면 False 출력

In [20]:
print(3 >= 4)

False


# 107
아래 코드의 출력 결과를 예상하라

if 4 < 3:
    
    print("Hello World")

답 : 아무것도 출력 안함

In [19]:
if 4 < 3:
    print("Hello World")

# 108
아래 코드의 출력 결과를 예상하라

if 4 < 3:

    print("Hello World.")

else:

    print("Hi, there.")

답 : Hi, there. 출력

In [21]:
if 4 < 3:
    print("Hello World.")
else:
    print("Hi, there.")

Hi, there.


# 109
아래 코드의 출력 결과를 예상하라

if True :
    
    print ("1")
    
    print ("2")

else :
    
    print("3")

print("4")

답 :

1

2

4

출력

In [22]:
if True :
    print ("1")
    print ("2")
else :
    print("3")
print("4")

1
2
4


# 110
아래 코드의 출력 결과를 예상하라

if True :
    
    if False:
        
        print("1")
        
        print("2")
    
    else:
        
        print("3")

else :
    
    print("4")

print("5")

답 :

3

5

출력

In [23]:
if True :
    if False:
        print("1")
        print("2")
    else:
        print("3")
else :
    print("4")
print("5")

3
5


# 111
사용자로부터 입력받은 문자열을 두 번 출력하라. 아래는 사용자가 "안녕하세요"를 입력한 경우의 출력 결과이다.

In [24]:
a=input()
print(a*2)

안녕하세요
안녕하세요안녕하세요


# 112
사용자로부터 하나의 숫자를 입력받고, 입력 받은 숫자에 10을 더해 출력하라.

In [28]:
a=int(input("숫자를 입력하세요 : "))
print(a+10)

숫자를 입력하세요 : 30
40


# 113
사용자로부터 하나의 숫자를 입력 받고 짝수/홀수를 판별하라.

In [33]:
a = int(input())
if a%2==0:
    print("짝수")
else :
    print("홀수")

30
짝수


# 114
사용자로부터 값을 입력받은 후 해당 값에 20을 더한 값을 출력하라. 단 사용자가 입력한 값과 20을 더한 계산 값이 255를 초과하는 경우 255를 출력해야 한다.

In [38]:
a = int(input("입력값 : "))
a+=20
if a>255:
    print('출력값 : 255')
else:
    print(f'출력값 : {a}')

입력값 : 240
출력값 : 255


# 115
사용자로부터 하나의 값을 입력받은 후 해당 값에 20을 뺀 값을 출력하라. 단 출력 값의 범위는 0~255이다. 예를 들어 결괏값이 0보다 작은 값이되는 경우 0을 출력하고 255보다 큰 값이 되는 경우 255를 출력해야 한다.

In [40]:
a = int(input())
a-=20
if a>255:
    print('출력값 : 255')
elif a<0:
    print('출력값 : 0')
else:
    print(f'출력값 : {a}')

15
출력값 : 0


# 116
사용자로부터 입력 받은 시간이 정각인지 판별하라.

In [44]:
a=input("현재시간:")
ans=a[-2:]
if ans =='00':
    print("정각 입니다")
else:
    print("정각이 아닙니다")

현재시간:03:10
정각이 아닙니다


# 117
사용자로 입력받은 단어가 아래 fruit 리스트에 포함되어 있는지를 확인하라. 포함되었다면 "정답입니다"를 아닐 경우 "오답입니다" 출력하라.

fruit = ["사과", "포도", "홍시"]

In [47]:
fruit = ["사과", "포도", "홍시"]
a=input("좋아하는 과일은? ")
if a in fruit:
    print("정답입니다.")
else:
    print("오답입니다")

좋아하는 과일은? 사과
정답입니다.


# 118
투자 경고 종목 리스트가 있을 때 사용자로부터 종목명을 입력 받은 후 해당 종목이 투자 경고 종목이라면 '투자 경고 종목입니다'를 아니면 "투자 경고 종목이 아닙니다."를 출력하는 프로그램을 작성하라.

warn_investment_list = ["Microsoft", "Google", "Naver", "Kakao", "SAMSUNG", "LG"]

In [49]:
warn_investment_list = ["Microsoft", "Google", "Naver", "Kakao", "SAMSUNG", "LG"]
a=input("투자 종목 입력 : ")
if a in warn_investment_list:
    print("투자 경고 종목입니다")
else:
    print("투자 경고 종목이 아닙니다")

투자 종목 입력 : Microsoft
투자 경고 종목입니다


# 119
아래와 같이 fruit 딕셔너리가 정의되어 있다. 사용자가 입력한 값이 딕셔너리 키 (key) 값에 포함되었다면 "정답입니다"를 아닐 경우 "오답입니다" 출력하라.

fruit = {"봄" : "딸기", "여름" : "토마토", "가을" : "사과"}

In [50]:
fruit = {"봄" : "딸기", "여름" : "토마토", "가을" : "사과"}
a=input("제가좋아하는계절은 : ")
if a in fruit:
    print("정답입니다.")
else:
    print("오답입니다.")

제가좋아하는계절은 : 봄
정답입니다.


# 120
아래와 같이 fruit 딕셔너리가 정의되어 있다. 사용자가 입력한 값이 딕셔너리 값 (value)에 포함되었다면 "정답입니다"를 아닐 경우 "오답입니다" 출력하라.

fruit = {"봄" : "딸기", "여름" : "토마토", "가을" : "사과"}

In [53]:
fruit = {"봄" : "딸기", "여름" : "토마토", "가을" : "사과"}
a=input("좋아하는과일은? ")
if a in fruit.values():
    print("정답입니다.")
else:
    print("오답입니다.")

좋아하는과일은? 한라봉
오답입니다.


# 121
사용자로부터 문자 한 개를 입력 받고, 소문자일 경우 대문자로, 대문자 일 경우, 소문자로 변경해서 출력하라.

In [54]:
a=input()
if a.islower():
    print(a.upper())
else:
    print(a.lower())

a
A


# 122
점수 구간에 해당하는 학점이 아래와 같이 정의되어 있다. 사용자로부터 score를 입력받아 학점을 출력하라.

In [55]:
score = int(input("score: "))
if score>80:
    print("grade is A")
elif score>60:
    print("grade is B")
elif score>40:
    print("grade is C")
elif score>20:
    print("grade is D")
else:
    print("grade is E")

score: 83
grade is A


# 123
사용자로부터 달러, 엔, 유로, 또는 위안 금액을 입력받은 후 이를 원으로 변환하는 프로그램을 작성하라. 각 통화별 환율은 다음과 같다. 사용자는 100 달러, 1000 엔, 13 유로, 100 위안과 같이 금액과 통화명 사이에 공백을 넣어 입력한다고 가정한다.

In [62]:
dic = {"달러": 1167, "엔": 1.096, "유로": 1268, "위안": 171}
a,b = input("입력 : ").split()
a=int(a)
ans = float(a*dic[b])
print(ans,'원')

입력 : 100 달러
116700.0 원


# 124
사용자로부터 세 개의 숫자를 입력 받은 후 가장 큰 숫자를 출력하라.

In [63]:
ans=[]
for i in range(3):
    k=int(input(f'input number{i+1}: '))
    ans.append(k)
print(max(ans))

input number1: 10
input number2: 9
input number3: 20
20


# 125
휴대폰 번호 앞자리에 따라 통신사는 아래와 같이 구분된다. 사용자로부터 휴대전화 번호를 입력 받고, 통신사를 출력하는 프로그램을 작성하라.

In [66]:
a=input("휴대전화 번호 입력: ")
a=a[:3]
dic={'011':'SKT','016':'KT','019':'LGU','010':'알수없음'}
ans=dic[a]
print(f'당신은 {ans} 사용자입니다.')

휴대전화 번호 입력: 011-345-1922
당신은 SKT 사용자입니다.


# 126
우편번호는 5자리로 구성되는데, 앞의 세자리는 구를 나타낸다. 예를들어, 강북구의 경우 010, 011, 012 세 자리로 시작한다.

In [70]:
a = input("우편번호: ")
a = int(a[2])
if a in [0,1,2]:
    print("강북구")
elif a in [3,4,5]:
    print("도봉구")
else:
    print("노원구")

우편번호: 01400
도봉구


# 127
주민등록번호 뒷 자리 7자리 중 첫째 자리는 성별을 나타내는데, 1, 3은 남자 2, 4는 여자를 의미한다. 사용자로부터 13자리의 주민등록번호를 입력 받은 후 성별 (남자, 여자)를 출력하는 프로그램을 작성하라.

In [76]:
a,b=input('주민등록번호: ').split('-')
ans=int(b[0])
if ans in [1,3]:
    print("남자")
else:
    print("여자")

주민등록번호: 821010-1635210
남자


# 128
주민등록번호의 뒷 자리 7자리 중 두번째와 세번째는 지역코드를 의미한다. 주민 등록 번호를 입력 받은 후 출생지가 서울인지 아닌지 판단하는 코드를 작성하라

In [80]:
a,b=input('주민등록번호: ').split('-')
ans = b[1:3]
if ans in ['00','01','02','03','04','05','06','07','08']:
    print("서울 입니다")
else:
    print("서울이 아닙니다")

주민등록번호: 861010-1015210
서울 입니다


# 129
주민등록번호는 13자리로 구성되는데 마지막 자리수는 주민등록번호의 유효성을 체크하는데 사용된다. 먼저 앞에서부터 12자리의 숫자에 2, 3, 4, 5, 6, 7, 8, 9, 2, 3, 4, 5를 차례로 곱한 뒤 그 값을 전부 더한다. 연산 결과 값을 11로 나누면 나머지가 나오는데 11에서 나머지를 뺀 값이 주민등록번호의 마지막 번호가 된다.

In [87]:
a,b=input('주민등록번호: ').split('-')
first = 0
for i in range(6):
    first += int(a[i])*(i+2)
for i in range(2):
    first += int(b[i])*(i+8)
for i in range(2,6):
    first += int(b[i])*i
first %= 11
second = 11-first
if second == b[-1]:
    print("유효한 주민등록번호입니다.")
else:
    print("유효하지 않은 주민등록번호입니다")

주민등록번호: 821010-1635210
유효하지 않은 주민등록번호입니다


# 130
아래 코드는 비트코인의 가격 정보를 딕셔너리로 가져오는 코드이다.

import requests

btc = requests.get("https://api.bithumb.com/public/ticker/").json()['data']

btc 딕셔너리 안에는 시가, 종가, 최고가, 최저가 등이 저장되어 있다. 최고가와 최저가의 차이를 변동폭으로 정의할 때 (시가 + 변동폭)이 최고가 보다 높을 경우 "상승장", 그렇지 않은 경우 "하락장" 문자열을 출력하라.

In [90]:
import requests
btc = requests.get("https://api.bithumb.com/public/ticker/").json()['data']

opening_price=float(btc['opening_price'])
closing_price=float(btc['closing_price'])
min_price=float(btc['min_price'])
max_price=float(btc['max_price'])

a = opening_price + (max_price - min_price)

if a > max_price:
    print("상승장")
else:
    print("하락장")

상승장


# <08. 파이썬 반복문>

# 131
for문의 실행결과를 예측하라.

과일 = ["사과", "귤", "수박"]

for 변수 in 과일:

    print(변수)

답 :

사과

귤

수박

출력

In [91]:
과일 = ["사과", "귤", "수박"]
for 변수 in 과일:
    print(변수)

사과
귤
수박


# 132
for문의 실행결과를 예측하라.

과일 = ["사과", "귤", "수박"]

for 변수 in 과일:

    print("#####")

답:

####

####

####

출력

In [93]:
과일 = ["사과", "귤", "수박"]
for 변수 in 과일:
    print("#####")

#####
#####
#####


# 133
다음 for 문과 동일한 기능을 수행하는 코드를 작성하세요.

for 변수 in ["A", "B", "C"]:
  
     print(변수)

In [95]:
print("A")
print("B")
print("C")

A
B
C


# 134
for문을 풀어서 동일한 동작을하는 코드를 작성하라.

for 변수 in ["A", "B", "C"]:
  
    print("출력:", 변수)

In [96]:
print("출력: A")
print("출력: B")
print("출력: C")

출력: A
출력: B
출력: C


In [97]:
for 변수 in ["A", "B", "C"]:
  print("출력:", 변수)

출력: A
출력: B
출력: C


# 135
for문을 풀어서 동일한 동작을 하는 코드를 작성하라.

for 변수 in ["A", "B", "C"]:

    b = 변수.lower()

    print("변환:", b)

In [102]:
print(f"변환: {'A'.lower()}")
print(f"변환: {'B'.lower()}")
print(f"변환: {'C'.lower()}")

변환: a
변환: b
변환: c


In [98]:
for 변수 in ["A", "B", "C"]:
  b = 변수.lower()
  print("변환:", b)

변환: a
변환: b
변환: c


# 136
다음 코드를 for문으로 작성하라.

변수 = 10

print(변수)

변수 = 20

print(변수)

변수 = 30

print(변수)

In [104]:
for i in range(3):
    print(int(input("변수 = ")))

변수 = 10
10
변수 = 20
20
변수 = 30
30


# 137
다음 코드를 for문으로 작성하라.

print(10)

print(20)

print(30)

In [105]:
for i in range(3):
    print(10*(i+1))

10
20
30


# 138
다음 코드를 for문으로 작성하라.

print(10)

print("-------")

print(20)

print("-------")

print(30)

print("-------")

In [108]:
for i in range(3):
    print(10*(i+1))
    print('-------')

10
-------
20
-------
30
-------


# 139
다음 코드를 for문으로 작성하라.

print("++++")

print(10)

print(20)

print(30)

In [107]:
print("++++")
for i in range(3):
    print(10*(i+1))

++++
10
20
30


# 140
다음 코드를 for문으로 작성하라.

print("-------")

print("-------")

print("-------")

print("-------")

In [109]:
for i in range(4):
    print("-------")

-------
-------
-------
-------


# 141
다음과 같이 판매가가 저장된 리스트가 있을 때 부가세가 포함된 가격을 for 문을 사용해서 화면에 출력하라. 단 부가세는 10원으로 가정한다.

리스트 = [100, 200, 300]

110

210

310

In [110]:
리스트 = [100,200,300]
for i in range(3):
    print(리스트[i]+10)

110
210
310


# 142
for 문을 사용해서 리스트에 저장된 값을 다음과 같이 출력하라.

리스트 = ["김밥", "라면", "튀김"]

오늘의 메뉴: 김밥

오늘의 메뉴: 라면

오늘의 메뉴: 튀김

In [111]:
리스트 = ["김밥", "라면", "튀김"]
for i in range(3):
    print(f'오늘의 메뉴: {리스트[i]}')

오늘의 메뉴: 김밥
오늘의 메뉴: 라면
오늘의 메뉴: 튀김


# 143
리스트에 주식 종목이름이 저장돼 있다.

리스트 = ["SK하이닉스", "삼성전자", "LG전자"]

저장된 문자열의 길이를 다음과 같이 출력하라.

6

4

4

In [112]:
리스트 = ["SK하이닉스", "삼성전자", "LG전자"]
for i in range(3):
    print(len(리스트[i]))

6
4
4


# 144
리스트에는 동물이름이 문자열로 저장돼 있다.

리스트 = ['dog', 'cat', 'parrot']

동물 이름과 글자수를 다음과 같이 출력하라.

dog 3

cat 3

parrot 6

In [113]:
리스트 = ['dog', 'cat', 'parrot']
for i in range(3):
    print(리스트[i],end=' ')
    print(len(리스트[i]))

dog 3
cat 3
parrot 6


# 145
리스트에 동물 이름 저장돼 있다.

리스트 = ['dog', 'cat', 'parrot']

for문을 사용해서 동물 이름의 첫 글자만 출력하라.

d

c

p

In [114]:
리스트 = ['dog', 'cat', 'parrot']
for i in range(3):
    print(리스트[i][0])

d
c
p


# 146
리스트에는 세 개의 숫자가 바인딩돼 있다.

리스트 = [1, 2, 3]

for문을 사용해서 다음과 같이 출력하라.

3 x 1

3 x 2

3 x 3

In [115]:
리스트 = [1,2,3]
for i in range(3):
    print(f'3 X {리스트[i]}')

3 X 1
3 X 2
3 X 3


# 147
리스트에는 세 개의 숫자가 바인딩돼 있다.

리스트 = [1, 2, 3]

for문을 사용해서 다음과 같이 출력하라.

3 x 1 = 3

3 x 2 = 6

3 x 3 = 9

In [116]:
리스트 = [1,2,3]
리스트 = [1,2,3]
for i in range(3):
    print(f'3 X {리스트[i]} = {3*리스트[i]}')

3 X 1 = 3
3 X 2 = 6
3 X 3 = 9


# 148
리스트에는 네 개의 문자열이 바인딩돼 있다.

리스트 = ["가", "나", "다", "라"]

for문을 사용해서 다음과 같이 출력하라.

나

다

라

In [123]:
리스트 = ["가", "나", "다", "라"]
del 리스트[0]
for i in range(len(리스트)):
    print(리스트[i])

나
다
라


# 149
리스트에는 네 개의 문자열이 바인딩돼 있다.

리스트 = ["가", "나", "다", "라"]

for문을 사용해서 다음과 같이 출력하라.

가

다

In [126]:
리스트 = ["가", "나", "다", "라"]
for i in range(2):
    print(리스트[i*2])

가
다


# 150
리스트에는 네 개의 문자열이 바인딩돼 있다.

리스트 = ["가", "나", "다", "라"]

for문을 사용해서 다음과 같이 출력하라.

라

다

나

가

In [129]:
리스트 = ["가", "나", "다", "라"]
리스트.reverse()
for i in range(4):
    print(리스트[i])

라
다
나
가


# 151
리스트에는 네 개의 정수가 저장돼 있다.

리스트 = [3, -20, -3, 44]

for문을 사용해서 리스트의 음수를 출력하라.

-20

-3

In [130]:
리스트 = [3, -20, -3, 44]
for i in 리스트:
    if i<0:
        print(i)

-20
-3


# 152
for문을 사용해서 3의 배수만을 출력하라.

리스트 = [3, 100, 23, 44]

3

In [131]:
리스트 = [3, 100, 23, 44]
for i in 리스트:
    if i%3==0:
        print(i)

3


# 153
리스트에서 20 보다 작은 3의 배수를 출력하라

리스트 = [13, 21, 12, 14, 30, 18]

12

18

In [136]:
리스트 = [13, 21, 12, 14, 30, 18]
for i in 리스트:
    if i<20 and i%3==0:
        print(i)

12
18


# 154
리스트에서 세 글자 이상의 문자를 화면에 출력하라

리스트 = ["I", "study", "python", "language", "!"]

study

python

language

In [143]:
리스트 = ["I", "study", "python", "language", "!"]
for i in range(len(리스트)):
    a = 리스트[i]
    if len(a)>=3:
        print(리스트[i])

study
python
language


# 155
리스트에서 대문자만 화면에 출력하라.

리스트 = ["A", "b", "c", "D"]

A

D

In [144]:
리스트 = ["A", "b", "c", "D"]
for i in 리스트:
    if i.isupper():
        print(i)

A
D


# 156
리스트에서 소문자만 화면에 출력하라.

리스트 = ["A", "b", "c", "D"]

b

c

In [145]:
리스트 = ["A", "b", "c", "D"]
for i in 리스트:
    if i.islower():
        print(i)

b
c


# 157
이름의 첫 글자를 대문자로 변경해서 출력하라.

리스트 = ['dog', 'cat', 'parrot']

Dog

Cat

Parrot

In [151]:
리스트 = ['dog', 'cat', 'parrot']
for i in range(3):
    a=리스트[i][0]
    b=리스트[i][1:]
    k=a.upper()
    print(k+b)

Dog
Cat
Parrot


# 158
파일 이름이 저장된 리스트에서 확장자를 제거하고 파일 이름만 화면에 출력하라. (힌트: split() 메서드)

리스트 = ['hello.py', 'ex01.py', 'intro.hwp']

hello

ex01

intro

In [152]:
리스트 = ['hello.py', 'ex01.py', 'intro.hwp']
for i in 리스트:
    ans=i.split(".")
    print(ans[0])

hello
ex01
intro


# 159
파일 이름이 저장된 리스트에서 확장자가 .h인 파일 이름을 출력하라.

리스트 = ['intra.h', 'intra.c', 'define.h', 'run.py']

intra.h

define.h

In [154]:
리스트 = ['intra.h', 'intra.c', 'define.h', 'run.py']
for i in 리스트:
    if '.h' in i:
        print(i)

intra.h
define.h


# 160
파일 이름이 저장된 리스트에서 확장자가 .h나 .c인 파일을 화면에 출력하라.

리스트 = ['intra.h', 'intra.c', 'define.h', 'run.py']

intra.h

intra.c

define.h

In [155]:
리스트 = ['intra.h', 'intra.c', 'define.h', 'run.py']
for i in 리스트:
    if '.h' in i or '.c' in i:
        print(i)

intra.h
intra.c
define.h


# 161
for문과 range 구문을 사용해서 0~99까지 한 라인에 하나씩 순차적으로 출력하는 프로그램을 작성하라.

In [156]:
for i in range(100):
    print(i)

0
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99


# 162
월드컵은 4년에 한 번 개최된다. range()를 사용하여 2002~2050년까지 중 월드컵이 개최되는 연도를 출력하라.

2002

2006

2010

...

2042

2046

2050

In [158]:
for i in range(13):
    print(2002+(i*4))

2002
2006
2010
2014
2018
2022
2026
2030
2034
2038
2042
2046
2050


# 163
1부터 30까지의 숫자 중 3의 배수를 출력하라.

In [159]:
for i in range(3,33,3):
    print(i)

3
6
9
12
15
18
21
24
27
30


# 164
99부터 0까지 1씩 감소하는 숫자들을, 한 라인에 하나씩 출력하라.

In [161]:
for i in range(99,-1,-1):
    print(i)

99
98
97
96
95
94
93
92
91
90
89
88
87
86
85
84
83
82
81
80
79
78
77
76
75
74
73
72
71
70
69
68
67
66
65
64
63
62
61
60
59
58
57
56
55
54
53
52
51
50
49
48
47
46
45
44
43
42
41
40
39
38
37
36
35
34
33
32
31
30
29
28
27
26
25
24
23
22
21
20
19
18
17
16
15
14
13
12
11
10
9
8
7
6
5
4
3
2
1
0


# 165
for문을 사용해서 아래와 같이 출력하라.

0.0

0.1

0.2

0.3

0.4

0.5

...

0.9

In [162]:
for i in range(10):
    print(i/10)

0.0
0.1
0.2
0.3
0.4
0.5
0.6
0.7
0.8
0.9


# 166
구구단 3단을 출력하라.

In [163]:
for i in range(10):
    print(f'3x{i+1} = {3*(i+1)}')

3x1 = 3
3x2 = 6
3x3 = 9
3x4 = 12
3x5 = 15
3x6 = 18
3x7 = 21
3x8 = 24
3x9 = 27
3x10 = 30


# 167
구구단 3단을 출력하라. 단 홀수 번째만 출력한다.

In [164]:
for i in range(5):
    print(f'3x{2*i+1} = {3*(2*i+1)}')

3x1 = 3
3x3 = 9
3x5 = 15
3x7 = 21
3x9 = 27


# 168
1~10까지의 숫자에 대해 모두 더한 값을 출력하는 프로그램을 for 문을 사용하여 작성하라.

In [165]:
ans = 0
for i in range(1,11):
    ans+=i
print(ans)

55


# 169
1~10까지의 숫자 중 모든 홀수의 합을 출력하는 프로그램을 for 문을 사용하여 작성하라.

In [166]:
ans = 0
for i in range(1,11,2):
    ans+=i
print(ans)

25


# 170
1~10까지의 숫자를 모두 곱한 값을 출력하는 프로그램을 for 문을 사용하여 작성하라.

In [167]:
ans = 1
for i in range(1,11):
    ans*=i
print(ans)

3628800


# 171
아래와 같이 리스트의 데이터를 출력하라. 단, for문과 range문을 사용하라.

price_list = [32100, 32150, 32000, 32500]
32100

32150

32000

32500

In [168]:
price_list = [32100, 32150, 32000, 32500]
for i in range(4):
    print(price_list[i])

32100
32150
32000
32500


# 172
아래와 같이 리스트의 데이터를 출력하라. 단, for문과 range문을 사용하라.

price_list = [32100, 32150, 32000, 32500]

0 32100

1 32150

2 32000

3 32500

In [169]:
price_list = [32100, 32150, 32000, 32500]
for i in range(4):
    print(i,end=' ')
    print(price_list[i])

0 32100
1 32150
2 32000
3 32500


# 173
아래와 같이 리스트의 데이터를 출력하라. 단, for문과 range문을 사용하라.

price_list = [32100, 32150, 32000, 32500]

3 32100

2 32150

1 32000

0 32500

In [170]:
price_list = [32100, 32150, 32000, 32500]
for i in range(4):
    print(3-i,end=' ')
    print(price_list[i])

3 32100
2 32150
1 32000
0 32500


# 174
아래와 같이 리스트의 데이터를 출력하라. 단, for문과 range문을 사용하라.

price_list = [32100, 32150, 32000, 32500]

100 32150

110 32000

120 32500

In [172]:
price_list = [32100, 32150, 32000, 32500]
del price_list[0]
for i in range(3):
    print(100+i*10,end=' ')
    print(price_list[i])

100 32150
110 32000
120 32500


# 175
my_list를 아래와 같이 출력하라.

my_list = ["가", "나", "다", "라"]

가 나

나 다

다 라

In [173]:
my_list = ["가", "나", "다", "라"]
for i in range(3):
    print(my_list[i],my_list[i+1])

가 나
나 다
다 라


# 176
리스트를 아래와 같이 출력하라.

my_list = ["가", "나", "다", "라", "마"]

가 나 다

나 다 라

다 라 마

In [174]:
my_list = ["가", "나", "다", "라", "마"]
for i in range(3):
    print(my_list[i],my_list[i+1],my_list[i+2])

가 나 다
나 다 라
다 라 마


# 177
반복문과 range 함수를 사용해서 my_list를 아래와 같이 출력하라.

my_list = ["가", "나", "다", "라"]

라 다

다 나

나 가

In [175]:
my_list = ["가", "나", "다", "라"]
for i in range(3):
    print(my_list[3-i],my_list[2-i])

라 다
다 나
나 가


# 178
리스트에는 네 개의 정수가 저장되어 있다. 각각의 데이터에 대해서 자신과 우측값과의 차분값을 화면에 출력하라.

my_list = [100, 200, 400, 800]

예를들어 100을 기준으로 우측에 위치한 200과의 차분 값를 화면에 출력하고, 200을 기준으로 우측에 위치한 400과의 차분값을 화면에 출력한다. 이어서 400을 기준으로 우측에 위치한 800과의 차분값을 화면에 출력한다.

100

200

400

In [176]:
my_list = [100, 200, 400, 800]
for i in range(3):
    print(my_list[i+1]-my_list[i])

100
200
400


# 179
리스트에는 6일 간의 종가 데이터가 저장되어 있다. 종가 데이터의 3일 이동 평균을 계산하고 이를 화면에 출력하라.

my_list = [100, 200, 400, 800, 1000, 1300]

첫 번째 줄에는 100, 200, 400의 평균값이 출력된다. 두 번째 줄에는 200, 400, 800의 평균값이 출력된다. 같은 방식으로 나머지 데이터의 평균을 출력한다.


233.33333333333334

466.6666666666667

733.3333333333334

1033.3333333333333

In [177]:
my_list = [100, 200, 400, 800, 1000, 1300]
for i in range(4):
    x=my_list[i]+my_list[i+1]+my_list[i+2]
    x/=3
    print(x)

233.33333333333334
466.6666666666667
733.3333333333334
1033.3333333333333


# 180
리스트에 5일간의 저가, 고가 정보가 저장돼 있다. 고가와 저가의 차를 변동폭이라고 정의할 때, low, high 두 개의 리스트를 사용해서 5일간의 변동폭을 volatility 리스트에 저장하라.

low_prices  = [100, 200, 400, 800, 1000]

high_prices = [150, 300, 430, 880, 1000]

In [180]:
low_prices  = [100, 200, 400, 800, 1000]
high_prices = [150, 300, 430, 880, 1000]
volatility=[]
for i in range(5):
    k=high_prices[i] - low_prices[i]
    volatility.append(k)
print(volatility)

[50, 100, 30, 80, 0]


# 181
아래 표에서 하나의 행을 하나의 리스트로, 총 3개의 리스트를 갖는 이차원 리스트 apart를 정의하라.

In [182]:
apart = [ ["101호", "102호"], ["201호", "202호"], ["301호", "302호"] ]
print(apart)

[['101호', '102호'], ['201호', '202호'], ['301호', '302호']]


# 182
아래 표에서 하나의 열을 하나의 리스트로, 총 2개의 리스트를 갖는 이차원 리스트 stock을 정의하라.

In [183]:
stock = [ ["시가", 100, 200, 300], ["종가", 80, 210, 330] ]
print(stock)

[['시가', 100, 200, 300], ['종가', 80, 210, 330]]


# 183
아래 표를 stock 이름의 딕셔너리로 표현하라.시가를 key로 저장하고, 나머지 같은 열의 데이터를 리스트로 저장해서 value로 저장한다. 종가 역시 key로 저장하고 나머지 같은 열의 데이터를 리스트로 저장해서 value로 저장한다.

In [184]:
stock = {"시가": [100, 200, 300], "종가": [80, 210, 330] }
print(stock)

{'시가': [100, 200, 300], '종가': [80, 210, 330]}


# 184
아래 표를 stock 이라는 이름의 딕셔너리로 표현하라. 날짜를 key로 저장하고, 나머지 같은 행의 데이터를 리스트로 저장해서 value로 저장한다. 첫 열이 날짜이다.

In [185]:
stock = {"10/10": [80, 110, 70, 90], "10/11": [210, 230, 190, 200] }
print(stock)

{'10/10': [80, 110, 70, 90], '10/11': [210, 230, 190, 200]}


# 185
리스트에 저장된 데이터를 아래와 같이 출력하라.

apart = [ [101, 102], [201, 202], [301, 302] ]

101 호


102 호

201 호

202 호

301 호

302 호

In [187]:
apart = [ [101, 102], [201, 202], [301, 302] ]
for i in range(3):
    for j in range(2):
        print(apart[i][j],'호')

101 호
102 호
201 호
202 호
301 호
302 호


# 186
리스트에 저장된 데이터를 아래와 같이 출력하라.

apart = [ [101, 102], [201, 202], [301, 302] ]

301 호

302 호

201 호

202 호

101 호

102 호

In [188]:
apart = [ [101, 102], [201, 202], [301, 302] ]
for i in range(3):
    for j in range(2):
        print(apart[2-i][j],'호')

301 호
302 호
201 호
202 호
101 호
102 호


# 187
리스트에 저장된 데이터를 아래와 같이 출력하라.

apart = [ [101, 102], [201, 202], [301, 302] ]

302 호

301 호

202 호

201 호

102 호

101 호

In [189]:
apart = [ [101, 102], [201, 202], [301, 302] ]
for i in range(3):
    for j in range(2):
        print(apart[2-i][1-j],'호')

302 호
301 호
202 호
201 호
102 호
101 호


# 188
리스트에 저장된 데이터를 아래와 같이 출력하라.

apart = [ [101, 102], [201, 202], [301, 302] ]

#### 101 호
#### -----
#### 102 호
#### -----
#### 201 호
#### -----
#### 202 호
#### -----
#### 301 호
#### -----
#### 302 호
#### -----

In [190]:
apart = [ [101, 102], [201, 202], [301, 302] ]
for i in range(3):
    for j in range(2):
        print(apart[i][j],'호')
        print('-----')

101 호
-----
102 호
-----
201 호
-----
202 호
-----
301 호
-----
302 호
-----


# 189
리스트에 저장된 데이터를 아래와 같이 출력하라.

apart = [ [101, 102], [201, 202], [301, 302] ]
#### 101 호
#### 102 호
#### -----
#### 201 호
#### 202 호
#### -----
#### 301 호
#### 302 호
#### -----

In [191]:
apart = [ [101, 102], [201, 202], [301, 302] ]
for i in range(3):
    for j in range(2):
        print(apart[i][j],'호')
    print('-----')

101 호
102 호
-----
201 호
202 호
-----
301 호
302 호
-----


# 190
리스트에 저장된 데이터를 아래와 같이 출력하라.

apart = [ [101, 102], [201, 202], [301, 302] ]

101 호

102 호

201 호

202 호

301 호

302 호
#### -----

In [192]:
apart = [ [101, 102], [201, 202], [301, 302] ]
for i in range(3):
    for j in range(2):
        print(apart[i][j],'호')
print('-----')

101 호
102 호
201 호
202 호
301 호
302 호
-----


# 191
ata에는 매수한 종목들의 OHLC (open/high/low/close) 가격 정보가 바인딩 되어있다.

data = [
    
    [ 2000,  3050,  2050,  1980],
    
    [ 7500,  2050,  2050,  1980],
    
    [15450, 15050, 15550, 14900]

]

수수료를 0.014 %로 가정할 때, 각 가격에 수수료를 포함한 가격을 한라인에 하나씩 출력하라.

2000.28

3050.427

2050.2870000000003

...


In [193]:
data = [
    [ 2000,  3050,  2050,  1980],
    [ 7500,  2050,  2050,  1980],
    [15450, 15050, 15550, 14900]
]
for i in range(3):
    for j in range(4):
        ans = data[i][j]
        x=ans*0.014
        x/=100
        print(ans+x)

2000.28
3050.427
2050.287
1980.2772
7501.05
2050.287
2050.287
1980.2772
15452.163
15052.107
15552.177
14902.086


# 192
191번의 출력 결과에 행단위로 "----" 구분자를 추가하라.

In [194]:
data = [
    [ 2000,  3050,  2050,  1980],
    [ 7500,  2050,  2050,  1980],
    [15450, 15050, 15550, 14900]
]
for i in range(3):
    for j in range(4):
        ans = data[i][j]
        x=ans*0.014
        x/=100
        print(ans+x)
    print('----')

2000.28
3050.427
2050.287
1980.2772
----
7501.05
2050.287
2050.287
1980.2772
----
15452.163
15052.107
15552.177
14902.086
----


# 193
192 번 문제의 결괏값을 result 이름의 리스트에 1차원 배열로 저장하라.

print(result)

[2000.28, 3050.427, 2050.2870000000003, 1980.2772, 7501.05, 2050.2870000000003, 2050.2870000000003, ...]

In [196]:
data = [
    [ 2000,  3050,  2050,  1980],
    [ 7500,  2050,  2050,  1980],
    [15450, 15050, 15550, 14900]
]
result=[]
for i in range(3):
    for j in range(4):
        ans = data[i][j]
        x=ans*0.014
        x/=100
        result.append(ans+x)
print(result)

[2000.28, 3050.427, 2050.287, 1980.2772, 7501.05, 2050.287, 2050.287, 1980.2772, 15452.163, 15052.107, 15552.177, 14902.086]


# 194
191번 문제의 결괏값을 result 이름의 리스트에 2차원 배열로 저장하라. 저장 포맷은 아래와 같다. 각 행에 대한 데이터끼리 리스트에 저장되어야 한다.

print(result)

[
 [2000.28, 3050.427, 2050.2870000000003, 1980.2772],
 [7501.05, 2050.2870000000003, 2050.2870000000003, 1980.2772],
 [15452.163, 15052.107, 15552.177, 14902.086000000001]
]

In [201]:
data = [
    [ 2000,  3050,  2050,  1980],
    [ 7500,  2050,  2050,  1980],
    [15450, 15050, 15550, 14900]
]
result=[]
for i in range(3):
    k=[]
    for j in range(4):
        ans = data[i][j]
        x=ans*0.014
        x/=100
        k.append(ans+x)
    result.append(k)
print(result)

[[2000.28, 3050.427, 2050.287, 1980.2772], [7501.05, 2050.287, 2050.287, 1980.2772], [15452.163, 15052.107, 15552.177, 14902.086]]


# 195
ohlc 리스트에는 시가(open), 고가 (high), 저가 (low) , 종가(close)가 날짜별로 저장돼 있다. 화면에 종가데이터를 출력하라.

ohlc = [["open", "high", "low", "close"],
        
        [100, 110, 70, 100],
        
        [200, 210, 180, 190],
        
        [300, 310, 300, 310]]

In [203]:
ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
for i in range(1,4):
    print(ohlc[i][-1])

100
190
310


# 196
ohlc 리스트에는 시가(open), 고가 (high), 저가 (low) , 종가(close)가 날짜별로 저장돼 있다. 종가가 150원보다 큰경우에만 종가를 출력하라.

ohlc = [["open", "high", "low", "close"],
        
        [100, 110, 70, 100],
        
        [200, 210, 180, 190],
        
        [300, 310, 300, 310]]

190

310

In [205]:
ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
for i in range(1,4):
    x=ohlc[i][-1]
    if x>150:
        print(x)

190
310


# 197
ohlc 리스트에는 시가(open), 고가 (high), 저가 (low) , 종가(close)가 날짜별로 저장돼 있다. 종가가 시가 보다 크거나 같은 경우에만 종가를 출력하라.

ohlc = [["open", "high", "low", "close"],
        
        [100, 110, 70, 100],
        
        [200, 210, 180, 190],
        
        [300, 310, 300, 310]]
100

310

In [206]:
ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
for i in range(1,4):
    if ohlc[i][-1] >= ohlc[i][0]:
        print(ohlc[i][-1])

100
310


# 198
ohlc 리스트에는 시가(open), 고가 (high), 저가 (low) , 종가(close)가 날짜별로 저장돼 있다. 고가와 저가의 차이를 변동폭으로 정의할 때 변동폭을 volatility 이름의 리스트에 저장하라.

ohlc = [["open", "high", "low", "close"],
        
        [100, 110, 70, 100],
        
        [200, 210, 180, 190],
        
        [300, 310, 300, 310]]

print(volatility)

[40, 30, 10]

In [207]:
ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
volatility=[]
for i in range(1,4):
    x=ohlc[i][1]-ohlc[i][2]
    volatility.append(x)
print(volatility)

[40, 30, 10]


# 199
리스트에는 3일 간의 ohlc 데이터가 저장돼 있다. 종가가 시가보다 높은 날의 변동성 (고가 - 저가)을 화면에 출력하라.

ohlc = [["open", "high", "low", "close"],
        
        [100, 110, 70, 100],
        
        [200, 210, 180, 190],
        
        [300, 310, 300, 310]]
        
종가가 시가보다 높은 거래일의 OHLC는 [300, 310, 300, 310] 이다. 따라서 이 거래일의 변동성은 10 (310 - 300)이다.

10

In [209]:
ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
for i in range(1,4):
    if ohlc[i][-1]>ohlc[i][0]:
        print(ohlc[i][1]-ohlc[i][2])

10


# 200
리스트에는 3일 간의 ohlc 데이터가 저장돼 있다. 시가에 매수해서 종가에 매도 했을 경우 총 수익금을 계산하라.

ohlc = [["open", "high", "low", "close"],
        
        [100, 110, 70, 100],
        
        [200, 210, 180, 190],
        
        [300, 310, 300, 310]]

1일차 수익 0원 (100 - 100), 2일차 수익 -10원 (190 - 200), 3일차 수익 10원 (310 - 300) 이다.

0

In [210]:
ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
ans=0
for i in range(1,4):
    ans += ohlc[i][0]-ohlc[i][-1]
print(ans)

0


# <09. 파이썬 함수>

# 201
"비트코인" 문자열을 화면에 출력하는 print_coin() 함수를 정의하라.

In [212]:
def print_coin():
    print("비트코인")

# 202
201번에서 정의한 함수를 호출하라.

In [213]:
print_coin()

비트코인


# 203
201번에서 정의한 print_coin 함수를 100번호출하라.

In [216]:
for i in range(100):
    print_coin()

비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인


# 204
"비트코인" 문자열을 100번 화면에 출력하는 print_coins() 함수를 정의하라.

In [215]:
def print_coins():
    for i in range(100):
        print("비트코인")
print_coins()

비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인


# 205
아래의 에러가 발생하는 이유에 대해 설명하라.

hello()

def hello():
    
    print("Hi")

답 : hello() 함수를 저장하기 전에 hello()를 실행해서 NameError 발생

# 206
아래 코드의 실행 결과를 예측하라.

def message() :
    
    print("A")
    
    print("B")

message()

print("C")

message()

답 :

A

B

C

A

B

출력

In [217]:
def message() :
    print("A")
    print("B")

message()
print("C")
message()

A
B
C
A
B


# 207
아래 코드의 실행 결과를 예측하라. (읽기 어려운 코드의 예입니다.)

print("A")

def message() :
    
    print("B")

print("C")

message()

답 :

A

C

B

출력

In [218]:
print("A")

def message() :
    print("B")

print("C")
message()

A
C
B


# 208
아래 코드의 실행 결과를 예측하라. (읽기 어려운 코드의 예입니다.)

print("A")

def message1() :
    
    print("B")

print("C")

def message2() :
    
    print("D")

message1()

print("E")

message2()

답 :

A

C

B

E

D

출력

In [219]:
print("A")
def message1() :
    print("B")
print("C")
def message2() :
    print("D")
message1()
print("E")
message2()

A
C
B
E
D


# 209
아래 코드의 실행 결과를 예측하라.

def message1():
    
    print("A")

def message2():
    
    print("B")
    
    message1()

message2()

답 :

B

A

출력

In [220]:
def message1():
    print("A")

def message2():
    print("B")
    message1()

message2()

B
A


# 210
아래 코드의 실행 결과를 예측하라.

def message1():
    
    print("A")

def message2():
    
    print("B")

def message3():
    
    for i in range (3) :
        
        message2()
        
        print("C")
    
    message1()

message3()

답 :

B

C

B

C

B

C

A

출력

In [221]:
def message1():
    print("A")

def message2():
    print("B")

def message3():
    for i in range (3) :
        message2()
        print("C")
    message1()

message3()

B
C
B
C
B
C
A


# 211
함수의 호출 결과를 예측하라.

def 함수(문자열) :
    
    print(문자열)

함수("안녕")

함수("Hi")

답:

안녕

HI

출력

In [222]:
def 함수(문자열) :
    print(문자열)

함수("안녕")
함수("Hi")

안녕
Hi


# 212
함수의 호출 결과를 예측하라.

def 함수(a, b) :
    
    print(a + b)

함수(3, 4)

함수(7, 8)

답:

7

15

출력

In [223]:
def 함수(a, b) :
    print(a + b)

함수(3, 4)
함수(7, 8)

7
15


# 213
아래와 같은 에러가 발생하는 원인을 설명하라.

def 함수(문자열) :
    
    print(문자열)

함수()

TypeError: 함수() missing 1 required positional argument: '문자열'

답 : 함수에 정의와 다르게 함수를 호출하고 있다. 함수를 호출할 때 하나의 파라미터를 입력해야한다.

# 214
아래와 같은 에러가 발생하는 원인을 설명하라.

def 함수(a, b) :
    
    print(a + b)

함수("안녕", 3)


답 : 문자열과 정수형은 더할 수 없음

# 215
하나의 문자를 입력받아 문자열 끝에 ":D" 스마일 문자열을 이어 붙여 출력하는 print_with_smile 함수를 정의하라.

In [228]:
def print_with_smile(string):
    print(string+":D")

# 216
215에서 정의한 함수를 호출하라. 파라미터는 "안녕하세요"로 입력하라

In [238]:
print_with_smile(input())

안녕하세요
안녕하세요:D


# 217
현재 가격을 입력 받아 상한가 (30%)를 출력하는 print_upper_price 함수를 정의하라.

In [237]:
def print_upper_price(x):
    print(x+x*0.3)
print_upper_price(int(input()))

100
130.0


# 218
두 개의 숫자를 입력받아 두 수의 합을 출력하는 print_sum 함수를 정의하라.

In [236]:
def print_sum(x,y):
    print(x+y)
a,b=map(int,input().split())
print_sum(a,b)

3 5
8


# 219
두 개의 숫자를 입력받아 합/차/곱/나눗셈을 출력하는 print_arithmetic_operation 함수를 작성하라.


print_arithmetic_operation(3, 4)

3 + 4 = 7

3 - 4 = -1

3 * 4 = 12

3 / 4 = 0.75


In [235]:
def print_arithmetic_operation(x,y):
    print(f'{x} + {y} = {x+y}')
    print(f'{x} - {y} = {x-y}')
    print(f'{x} * {y} = {x*y}')
    print(f'{x} / {y} = {x/y}')
a,b=map(int,input().split())
print_arithmetic_operation(a,b)

3 4
3 + 4 = 7
3 - 4 = -1
3 * 4 = 12
3 / 4 = 0.75


# 220
세 개의 숫자를 입력받아 가장 큰수를 출력하는 print_max 함수를 정의하라. 단 if 문을 사용해서 수를 비교하라.

In [244]:
def print_max(a,b,c):
    if a>b:
        if a>c:
            print(a)
        else:
            print(c)
    else:
        if b>c:
            print(b)
        else:
            print(c)
a,b,c=map(int,input().split())
print_max(a,b,c)

10 20 30
30


# 221
입력된 문자열을 역순으로 출력하는 print_reverse 함수를 정의하라.

print_reverse("python")

nohtyp

In [250]:
def print_reverse(string):
    print(string[::-1])
print_reverse(input())

python
nohtyp


# 222
성적 리스트를 입력 받아 평균을 출력하는 print_score 함수를 정의하라.

print_score ([1, 2, 3])

2.0

In [252]:
def print_score(list):
    print(sum(list)/len(list))
a=list(map(int,input().split()))
print_score(a)

1 2 3
2.0


# 223
하나의 리스트를 입력받아 짝수만 화면에 출력하는 print_even 함수를 정의하라.

print_even ([1, 3, 2, 10, 12, 11, 15])

2

10

12

In [258]:
def print_even(list):
    for i in list:
        if i %2 ==0:
            print(i)
a=list(map(int,input().split()))
print_even(a)

1 3 2 10 12 11 15
2
10
12


# 224
하나의 딕셔너리를 입력받아 딕셔너리의 key 값을 화면에 출력하는 print_keys 함수를 정의하라.

print_keys ({"이름":"김말똥", "나이":30, "성별":0})

이름

나이

성별

In [259]:
def print_keys(dic):
    for i in dic.keys():
        print(i)
print_keys ({"이름":"김말똥", "나이":30, "성별":0})

이름
나이
성별


# 225
my_dict에는 날짜를 키값으로 OHLC가 리스트로 저장돼 있다.

my_dict = {"10/26" : [100, 130, 100, 100],
           
           "10/27" : [10, 12, 10, 11]}

my_dict와 날짜 키값을 입력받아 OHLC 리스트를 출력하는 print_value_by_key 함수를 정의하라.

print_value_by_key  (my_dict, "10/26")

[100, 130, 100, 100]

In [260]:
my_dict = {"10/26" : [100, 130, 100, 100],
           "10/27" : [10, 12, 10, 11]}

def print_value_by_key(my_dict,key):
    print(my_dict[key])

print_value_by_key (my_dict, "10/26")

[100, 130, 100, 100]


# 226
입력 문자열을 한 줄에 다섯글자씩 출력하는 print_5xn(string) 함수를 작성하라.

print_5xn("아이엠어보이유알어걸")

아이엠어보

이유알어걸

In [268]:
def print_5xn(string):
    a = len(string)
    a//=5
    a+=1
    for i in range(a):
        print(string[i*5:(i+1)*5])
print_5xn(input())

아이엠어보이유알어걸
아이엠어보
이유알어걸



# 227
문자열과 한줄에 출력될 글자 수를 입력을 받아 한 줄에 입력된 글자 수만큼 출력하는 print_mxn(string) 함수를 작성하라.

printmxn("아이엠어보이유알어걸", 3)

아이엠

어보이

유알어

걸

In [272]:
def print_mxn(string,x):
    a = len(string)
    a//=x
    a+=1
    for i in range(a):
        print(string[i*x:(i+1)*x])
a=input()
b=int(input())
print_mxn(a,b)

아이엠어보이유알어걸
3
아이엠
어보이
유알어
걸


# 228
연봉을 입력받아 월급을 계산하는 calc_monthly_salary(annual_salary) 함수를 정의하라. 회사는 연봉을 12개월로 나누어 분할 지급하며, 이 때 1원 미만은 버림한다.

calc_monthly_salary(12000000)

1000000

In [274]:
def calc_monthly_salary(annual_salary):
    print(annual_salary//12)
calc_monthly_salary(int(input()))

12000000
1000000


# 229
아래 코드의 실행 결과를 예측하라.

def my_print (a, b) :
    
    print("왼쪽:", a)
    
    print("오른쪽:", b)

my_print(b=100, a=200)

답 :

왼쪽: 100

오른쪽: 200

출력

In [278]:
def my_print (a, b) :
    print("왼쪽:", a)
    print("오른쪽:", b)

my_print(a=100, b=200)

왼쪽: 100
오른쪽: 200


# 230
아래 코드의 실행 결과를 예측하라.

def my_print (a, b) :
    
    print("왼쪽:", a)
    
    print("오른쪽:", b)

my_print(b=100, a=200)


답 :

왼쪽: 100

오른쪽: 200

출력

In [277]:
def my_print (a, b) :
    print("왼쪽:", a)
    print("오른쪽:", b)

my_print(b=100, a=200)

왼쪽: 200
오른쪽: 100


# 231
아래 코드를 실행한 결과를 예상하라.

def n_plus_1 (n) :
    
    result = n + 1

n_plus_1(3)

print (result)

답 : 에러 발생

Since, 함수 내부에 있는 변수는 밖에서 접근 불가능

In [1]:
def n_plus_1 (n) :
    result = n + 1

n_plus_1(3)
print (result)

NameError: name 'result' is not defined

# 232
문자열 하나를 입력받아 인터넷 주소를 반환하는 make_url 함수를 정의하라.

make_url("naver")

www.naver.com

In [4]:
def make_url(string):
    print(f'www.{string}.com')
make_url("naver")

www.naver.com


# 233
문자열을 입력받아 각 문자들로 구성된 리스트로 반환하는 make_list 함수를 정의하라.

make_list("abcd")

['a', 'b', 'c', 'd']

In [9]:
def make_list(string):
    k=[]
    for i in range(len(string)):
        k.append(string[i])
    print(k)
make_list("abcd")

['a', 'b', 'c', 'd']


# 234
숫자로 구성된 하나의 리스트를 입력받아, 짝수들을 추출하여 리스트로 반환하는 

pickup_even 함수를 구현하라.

pickup_even([3, 4, 5, 6, 7, 8])

[4, 6, 8]

In [11]:
def pickup_even(list):
    ans=[]
    for i in list:
        if i%2==0:
            ans.append(i)
    print(ans)
pickup_even([3,4,5,6,7,8])

[4, 6, 8]


# 235
콤마가 포함된 문자열 숫자를 입력받아 정수로 변환하는 convert_int 함수를 정의하라.

convert_int("1,234,567")

1234567

In [14]:
def convert_int(string):
    k=string.replace(',','')
    k=int(k)
    print(k)
convert_int('1,234,567')

1234567


# 236
아래 코드의 실행 결과를 예측하라.

def 함수(num) :
    
    return num + 4

a = 함수(10)

b = 함수(a)

c = 함수(b)

print(c)

답 : 22

a=14 -> b=18 -> c=22



# 237
아래 코드의 실행 결과를 예측하라.

def 함수(num) :
    
    return num + 4

c = 함수(함수(함수(10)))

print(c)

답 : 22

(((10+4)+4)+4)+4=22

In [17]:
def 함수(num) :
    return num + 4

c = 함수(함수(함수(10)))
print(c)

22


# 238
아래 코드의 실행 결과를 예측하라.

def 함수1(num) :
    
    return num + 4

def 함수2(num) :
    
    return num * 10

a = 함수1(10)

c = 함수2(a)

print(c)

답 : 140

a = 14 -> c=140

In [18]:
def 함수1(num) :
    return num + 4

def 함수2(num) :
    return num * 10

a = 함수1(10)
c = 함수2(a)
print(c)

140


# 239
def 함수1(num) :
    
    return num + 4

def 함수2(num) :
    
    num = num + 2
    
    return 함수1(num)

c = 함수2(10)

print(c)

답 : 16

함수2(10)=함수1(12)=16

In [19]:
def 함수1(num) :
    return num + 4

def 함수2(num) :
    num = num + 2
    return 함수1(num)

c = 함수2(10)
print(c)

16


# 240
아래 코드의 실행 결과를 예측하라.

def 함수0(num) :
    
    return num * 2

def 함수1(num) :
    
    return 함수0(num + 2)

def 함수2(num) :
    
    num = num + 10
    
    return 함수1(num)

c = 함수2(2)

print(c)

답 : 28

함수2(2)=함수1(12)=함수0(14)=28

In [20]:
def 함수0(num) :
    return num * 2

def 함수1(num) :
    return 함수0(num + 2)

def 함수2(num) :
    num = num + 10
    return 함수1(num)

c = 함수2(2)
print(c)

28


# <파이썬 모듈>

# 241 현재시간

datetime 모듈을 사용해서 현재 시간을 화면에 출력해보세요.



In [21]:
import datetime
print(datetime.datetime.now())

2023-03-14 13:19:08.780821


# 242 현재시간의 타입
datetime 모듈의 now 함수의 리턴 값의 타입을 화면에 출력해보세요.

In [22]:
import datetime
ans=datetime.datetime.now()
print(ans,type(ans))

2023-03-14 13:19:58.904579 <class 'datetime.datetime'>


# 243 timedelta
datetime 모듈의 timedelta를 사용해서 오늘로부터 5일, 4일, 3일, 2일, 1일 전의 날짜를 화면에 출력해보세요.

In [23]:
import datetime

k = datetime.datetime.now()

for i in range(5,0,-1):
    delta = datetime.timedelta(days=i)
    ans = k-delta
    print(ans)

2023-03-09 13:21:23.053297
2023-03-10 13:21:23.053297
2023-03-11 13:21:23.053297
2023-03-12 13:21:23.053297
2023-03-13 13:21:23.053297


# 244 strftime
현재시간을 얻어온 후 다음과 같은 포맷으로 시간을 출력해보세요. strftime 메서드를 사용하세요.

In [24]:
import datetime
ans=datetime.datetime.now()
print(ans.strftime("%H:%M:%S"))

13:22:22


# 245 strptime
datetime.datetime.strptime 메서드를 사용하면 문자열 형식의 시간을 datetime.datetime 타입의 시간 값으로 만들어줍니다. "2020-05-04"의 문자열을 시간 타입으로 변환해보세요.

In [25]:
import datetime
a="2020-05-04"
ans = datetime.datetime.strptime(a,"%Y-%m-%d")
print(ans,type(ans))

2020-05-04 00:00:00 <class 'datetime.datetime'>


# 246 sleep 함수
time 모듈, datetime 모듈을 사용해서 1초에 한 번 현재 시간을 출력하는 코드를 작성하세요.

In [27]:
import time
import datetime
i = 0
while 1:
    if i==10:break
    ans = datetime.datetime.now()
    print(ans)
    time.sleep(1)
    i+=1

2023-03-14 13:25:14.869279
2023-03-14 13:25:15.871166
2023-03-14 13:25:16.885128
2023-03-14 13:25:17.892509
2023-03-14 13:25:18.894369
2023-03-14 13:25:19.897685
2023-03-14 13:25:20.900716
2023-03-14 13:25:21.904759
2023-03-14 13:25:22.907771
2023-03-14 13:25:23.911078


# 247 모듈임포트
모듈을 임포트하는 4가지 방식에 대해 설명해보세요.

답 :

1. import os              # os.rename()

2. from os import rename  # rename()

3. from os import *       # getcwd(),rename(),...

4. import os as myos      # myos.getcwd()

# 248 os모듈
os 모듈의 getcwd 함수를 호출하여 현재 디렉터리의 경로를 화면에 출력해보세요.

In [29]:
import os
ans = os.getcwd()
print(ans,type(ans))

C:\Users\kbk\Desktop\2023년1학기\MA <class 'str'>


# 249 rename 모듈
바탕화면에 텍스트 파일을 하나 생성한 후 os 모듈의 rename 함수를 호출하여 해당 파일의 이름을 변경해보세요.

# 250 numpy
numpy 모듈의 arange 함수를 사용해서 0.0 부터 5.0까지 0.1씩 증가하는 값을 화면에 출력해보세요.

In [41]:
import numpy as np
for i in np.arange(0,5.1,0.1):
    ans = round(i,2)
    print(ans)

0.0
0.1
0.2
0.3
0.4
0.5
0.6
0.7
0.8
0.9
1.0
1.1
1.2
1.3
1.4
1.5
1.6
1.7
1.8
1.9
2.0
2.1
2.2
2.3
2.4
2.5
2.6
2.7
2.8
2.9
3.0
3.1
3.2
3.3
3.4
3.5
3.6
3.7
3.8
3.9
4.0
4.1
4.2
4.3
4.4
4.5
4.6
4.7
4.8
4.9
5.0


# <파이썬 클래스>

# 251 클래스, 객체, 인스턴스
클래스, 객체, 인스턴스에 대해 설명해봅시다.

답:

클래스 : 똑같은 무언인가를 계속해서 만들어 낼 수 있는 설계 도면

객체 : 클래스로 만든 피조물

인스턴스 : 특정 객체가 어떤 클래스의 객체인지 괸계를 중점으로 표현할 때 사용

# 252 클래스 정의
비어있는 사람 (Human) 클래스를 "정의" 해보세요.

In [42]:
class Human:
    pass

# 253 인스턴스 생성
사람 (Human) 클래스의 인스턴스를 "생성" 하고 이를 areum 변수로 바인딩해보세요.

In [43]:
class Human:
    pass

areum = Human()

# 254 클래스 생성자-1
사람 (Human) 클래스에 "응애응애"를 출력하는 생성자를 추가하세요.

areum = Human()

응애응애

In [45]:
class Human():
    def __init__(self):
        print("응애응애")
areum = Human()

응애응애


# 255 클래스 생성자-2
사람 (Human) 클래스에 (이름, 나이, 성별)을 받는 생성자를 추가하세요.

areum = Human("아름", 25, "여자")


In [52]:
class Human:
    def __init__(self, name, age, sex):
        self.name = name
        self.age = age
        self.sex = sex

areum = Human("아름",25,"여자")
print(areum.name)
print(areum.age)
print(areum.sex)
print(areum)

아름
25
여자


# 256 인스턴스 속성에 접근
255에서 생성한 인스턴스의 이름, 나이, 성별을 출력하세요. 인스턴스 변수에 접근하여 값을 출력하면 됩니다.

이름: 조아름, 나이: 25, 성별: 여자

인스턴스 변수에 접근하여 값을 가져오는 예

areum.age

25

In [53]:
print(areum.name)
print(areum.age)
print(areum.sex)

아름
25
여자


# 257 클래스 메소드 - 1
사람 (Human) 클래스에서 이름, 나이, 성별을 출력하는 who() 메소드를 추가하세요.

areum.who()

이름: 조아름, 나이: 25, 성별: 여자


In [55]:
class Human:
    def __init__(self, name, age, sex):
        self.name = name
        self.age = age
        self.sex = sex
    
    def who(self):
        print("이름: {}, 나이: {}, 성별: {}".
             format(self.name,self.age,self.sex))
        
areum = Human("아름",25,"여자")
areum.who()

이름: 아름, 나이: 25, 성별: 여자


# 258 클래스 메소드 -2
사람 (Human) 클래스에 (이름, 나이, 성별)을 받는 setInfo 메소드를 추가하세요.

areum = Human("모름", 0, "모름")

areum.setInfo("아름", 25, "여자")

In [58]:
class Human:
    def __init__(self, name, age, sex):
        self.name = name
        self.age = age
        self.sex = sex
    
    def who(self):
        print("이름: {}, 나이: {}, 성별: {}".
             format(self.name,self.age,self.sex))
        
    def setInfo(self, name, age, sex):
        self.name = name
        self.age = age
        self.sex = sex

areum = Human("모름","모름","모름")
areum.who()
print("\n")
areum = Human("아름",25,"여자")
areum.who()

이름: 모름, 나이: 모름, 성별: 모름


이름: 아름, 나이: 25, 성별: 여자


# 259
사람 (human) 클래스에 "나의 죽음을 알리지 말라"를 출력하는 소멸자를 추가하세요.

areum = Human("아름", 25, "여자")

del areum

나의 죽음을 알리지 말라

In [59]:
class Human:
    def __init__(self, name, age, sex):
        self.name = name
        self.age = age
        self.sex = sex
    
    def who(self):
        print("이름: {}, 나이: {}, 성별: {}".
             format(self.name,self.age,self.sex))
        
    def setInfo(self, name, age, sex):
        self.name = name
        self.age = age
        self.sex = sex
    
    def __del__(self):
        print("나의 죽음을 알리지마라")

areum = Human("아름",25,"여자")
del(areum)

나의 죽음을 알리지마라


# 260 에러의 원인
아래와 같은 에러가 발생한 원인에 대해 설명하세요.

class OMG : 
    
    def print() :
        
        print("Oh my god")

myStock = OMG()

myStock.print()

TypeError Traceback (most recent call last)

<ipython-input-233-c85c04535b22> in <module>()

    ----> myStock.print()

TypeError: print() takes 0 positional arguments but 1 was given


In [60]:
class OMG : 
    def print() :
        print("Oh my god")

In [66]:
mystock = OMG()
OMG.print()

Oh my god


print()에 인자가 없는데

mystock.print()는 OMG.print(mystock)로 print 안에 인자가 들어서가서

에러가 뜬다

# 261 Stock 클래스 생성
주식 종목에 대한 정보를 저장하는 Stock 클래스를 정의해보세요. 클래스는 속성과 메서드를 갖고 있지 않습니다.


In [67]:
class Stock:
    pass

# 262 생성자
Stock 클래스의 객체가 생성될 때 종목명과 종목코드를 입력 받을 수 있도록 생성자를 정의해보세요.

삼성 = Stock("삼성전자", "005930")

In [69]:
class Stock:
    def __init__(self,name,code):
        self.name = name
        self.code = code

삼성 = Stock("삼성전자","005930")
print(삼성.name)
print(삼성.code)

삼성전자
005930


# 263 메서드
객체에 종목명을 입력할 수 있는 set_name 메서드를 추가해보세요.

a = Stock(None, None)

a.set_name("삼성전자")

In [70]:
class Stock:
    def __init__(self,name,code):
        self.name = name
        self.code = code
        
    def set_name(self,name):
        self.name = name

a=Stock(None,None)
a.set_name("삼성전자")
print(a.name)

삼성전자


# 264 메서드
객체에 종목코드를 입력할 수 있는 set_code 메서드를 추가해보세요.

a = Stock(None, None)

a.set_code("005930")

In [73]:
class Stock:
    def __init__(self,name,code):
        self.name = name
        self.code = code
        
    def set_name(self,name):
        self.name = name
        
    def set_code(self,code):
        self.code = code

a = Stock(None, None)
a.set_code("005930")
print(a.code)

005930


# 265 메서드
종목명과 종목코드를 리턴하는 get_name, get_code 메서드를 추가하세요. 해당 메서드를 사용하여 종목명과 종목코드를 얻고 이를 출력해보세요.

삼성 = Stock("삼성전자", "005930")

In [77]:
class Stock:
    def __init__(self,name,code):
        self.name = name
        self.code = code
        
    def set_name(self,name):
        self.name = name
        
    def set_code(self,code):
        self.code = code
        
    def get_name(self):
        return self.name
    
    def get_code(self):
        return self.code

삼성 = Stock("삼성전자","005930")
print(삼성.name)
print(삼성.code)
print("\n")
print(삼성.get_name())
print(삼성.get_code())

삼성전자
005930


삼성전자
005930


# 266 객체의 속성값 업데이트
생성자에서 종목명, 종목코드, PER, PBR, 배당수익률을 입력 받을 수 있도록 생성자를 수정하세요. PER, PBR, 배당수익률은 float 타입입니다.

In [79]:
class Stock:
    def __init__(self,name,code,PER,PBR,배당수익률):
        self.name = name
        self.code = code
        self.PER = PER
        self.PBR = PBR
        self.배당수익률 = 배당수익률
        
    def set_name(self,name):
        self.name = name
        
    def set_code(self,code):
        self.code = code
        
    def get_name(self):
        return self.name
    
    def get_code(self):
        return self.code

# 267 객체 생성
266번에서 정의한 생성자를 통해 다음 정보를 갖는 객체를 생성해보세요.

In [80]:
삼성 = Stock("삼성전자","005930",15.79,1.33,2.83)

# 268 객체의 속성 수정
PER, PBR, 배당수익률은 변경될 수 있는 값입니다. 이 값을 변경할 때 사용하는 set_per, set_pbr, set_dividend 메서드를 추가하세요.

In [83]:
class Stock:
    def __init__(self,name,code,per,pbr,dividend):
        self.name = name
        self.code = code
        self.per = per
        self.pbr = pbr
        self.dividend = dividend
        
    def set_name(self,name):
        self.name = name
        
    def set_code(self,code):
        self.code = code
        
    def get_name(self):
        return self.name
    
    def get_code(self):
        return self.code
    
    def set_per(self,per):
        self.per = per
        
    def set_pbr(self,pbr):
        self.pbr = pbr
    
    def set_dividend(self,dividend):
        self.dividend = dividend

# 269
267번에서 생성한 객체에 set_per 메서드를 호출하여 per 값을 12.75로 수정해보세요.

In [84]:
삼성 = Stock("삼성전자", "005930", 15.79, 1.33, 2.83)
삼성.set_per(12.75)
print(삼성.per)

12.75


# 270 여러 종목의 객체 생성
아래의 표를 참조하여 3종목에 대해 객체를 생성하고 이를 파이썬 리스트에 저장하세요. 파이썬 리스트에 저장된 각 종목에 대해 for 루프를 통해 종목코드와 PER을 출력해보세요.

In [85]:
종목 = []

삼성 = Stock("삼성전자", "005930", 15.79, 1.33, 2.83)
현대차 = Stock("현대차", "005380", 8.70, 0.35, 4.27)
LG전자 = Stock("LG전자", "066570", 317.34, 0.69, 1.37)

종목.append(삼성)
종목.append(현대차)
종목.append(LG전자)

for i in 종목:
    print(i.code, i.per)

005930 15.79
005380 8.7
066570 317.34


# 271 Account 클래스
은행에 가서 계좌를 개설하면 은행이름, 예금주, 계좌번호, 잔액이 설정됩니다. Account 클래스를 생성한 후 생성자를 구현해보세요. 생성자에서는 예금주와 초기 잔액만 입력 받습니다. 은행이름은 SC은행으로 계좌번호는 3자리-2자리-6자리 형태로 랜덤하게 생성됩니다.

은행이름: SC은행

계좌번호: 111-11-111111

In [2]:
import random
class Account:
    def __init__(self, name, balance):   # 예금주명, 초기잔액 입력
        self.name = name
        self.balance = balance
        self.bank = "SC은행"
        num1 = random.randint(0,999)
        num2 = random.randint(0,99)
        num3 = random.randint(0,999999)
        num1 = str(num1).zfill(3)   # 자릿수 맞추기 위해서 앞에 0으로 채우는 함수
        num2 = str(num2).zfill(2)
        num3 = str(num3).zfill(6)   
        self.account_number = num1+'-'+num2+'-'+num3

In [4]:
a = Account("A",10000)
print(a.name)
print(a.balance)
print(a.bank)
print(a.account_number)

A
10000
SC은행
369-99-163423


# 272 클래스 변수
클래스 변수를 사용해서 Account 클래스로부터 생성된 계좌 객체의 개수를 저장하세요.

In [7]:
class Account:
    count = 0
    def __init__(self, name, balance):   
        self.name = name
        self.balance = balance
        self.bank = "SC은행"
        num1 = random.randint(0,999)
        num2 = random.randint(0,99)
        num3 = random.randint(0,999999)
        num1 = str(num1).zfill(3)   
        num2 = str(num2).zfill(2)
        num3 = str(num3).zfill(6)   
        self.account_number = num1+'-'+num2+'-'+num3
        Account.count+=1
        
a1 = Account("A1",10000)
print(Account.count)
a2 = Account("A2",10000)
print(Account.count)
a3 = Account("A3",10000)
print(Account.count)
a4 = Account("A4",10000)
print(Account.count)
a5 = Account("A5",10000)
print(Account.count)
a6 = Account("A6",10000)
print(Account.count)

1
2
3
4
5
6


# 273 클래스 변수 출력
Account 클래스로부터 생성된 계좌의 개수를 출력하는 get_account_num() 메서드를 추가하세요.

In [10]:
class Account:
    count = 0
    def __init__(self, name, balance):   
        self.name = name
        self.balance = balance
        self.bank = "SC은행"
        num1 = random.randint(0,999)
        num2 = random.randint(0,99)
        num3 = random.randint(0,999999)
        num1 = str(num1).zfill(3)   
        num2 = str(num2).zfill(2)
        num3 = str(num3).zfill(6)   
        self.account_number = num1+'-'+num2+'-'+num3
        Account.count+=1
        
    def get_account_num(cls):
        print(cls.count)

a1 = Account("A1",10000)
a2 = Account("A2",10000)
a1.get_account_num()

2


# 274 입금 메서드
Account 클래스에 입금을 위한 deposit 메서드를 추가하세요. 입금은 최소 1원 이상만 가능합니다.

In [13]:
class Account:
    count = 0
    def __init__(self, name, balance):   
        self.name = name
        self.balance = balance
        self.bank = "SC은행"
        num1 = random.randint(0,999)
        num2 = random.randint(0,99)
        num3 = random.randint(0,999999)
        num1 = str(num1).zfill(3)   
        num2 = str(num2).zfill(2)
        num3 = str(num3).zfill(6)   
        self.account_number = num1+'-'+num2+'-'+num3
        Account.count+=1
        
    def get_account_num(cls):
        print(cls.count)
        
    def deposit(self, money_in):
        if money_in >=1:
            self.balance+=money_in

a1 = Account("A1",10000)
print(a1.balance)
a1.deposit(10000)
print(a1.balance)

10000
20000


# 275 출금메서드
Account 클래스에 출금을 위한 withdraw 메서드를 추가하세요. 출금은 계좌의 잔고 이상으로 출금할 수는 없습니다.

In [14]:
class Account:
    count = 0
    def __init__(self, name, balance):   
        self.name = name
        self.balance = balance
        self.bank = "SC은행"
        num1 = random.randint(0,999)
        num2 = random.randint(0,99)
        num3 = random.randint(0,999999)
        num1 = str(num1).zfill(3)   
        num2 = str(num2).zfill(2)
        num3 = str(num3).zfill(6)   
        self.account_number = num1+'-'+num2+'-'+num3
        Account.count+=1
        
    def get_account_num(cls):
        print(cls.count)
        
    def deposit(self, money_in):
        if money_in >=1:
            self.balance+=money_in
            
    def withdraw(self, money_out):
        if self.balance > money_out:
            self.balance-=money_out

a = Account("A1",10000)
print(a.balance)
a.deposit(5000)
print(a.balance)
a.withdraw(3000)
print(a.balance)

10000
15000
12000


# 276 정보 출력 메서드
Account 인스턴스에 저장된 정보를 출력하는 display_info() 메서드를 추가하세요. 잔고는 세자리마다 쉼표를 출력하세요.

은행이름: SC은행

예금주: 파이썬

계좌번호: 111-11-111111

잔고: 10,000원

In [17]:
class Account:
    count = 0
    def __init__(self, name, balance):   
        self.name = name
        self.balance = balance
        self.bank = "SC은행"
        num1 = random.randint(0,999)
        num2 = random.randint(0,99)
        num3 = random.randint(0,999999)
        num1 = str(num1).zfill(3)   
        num2 = str(num2).zfill(2)
        num3 = str(num3).zfill(6)   
        self.account_number = num1+'-'+num2+'-'+num3
        Account.count+=1
        
    def get_account_num(cls):
        print(cls.count)
        
    def deposit(self, money_in):
        if money_in >=1:
            self.balance+=money_in
            
    def withdraw(self, money_out):
        if self.balance > money_out:
            self.balance-=money_out
            
    def display_info(self):
        print(f"은행이름 : {self.bank}")
        print(f"예금주 : {self.name}")
        print(f"계좌번호 : {self.account_number}")
        print(f"잔고 : {self.balance}원")
        
a = Account("A1",10000)
a.display_info()

은행이름 : SC은행
예금주 : A1
계좌번호 : 422-15-896605
잔고 : 10000원


# 277 이자 지급하기
입금 횟수가 5회가 될 때 잔고를 기준으로 1%의 이자가 잔고에 추가되도록 코드를 변경해보세요.

In [23]:
class Account:
    count = 0
    def __init__(self, name, balance):   
        self.deposit_count = 0
        self.name = name
        self.balance = balance
        self.bank = "SC은행"
        num1 = random.randint(0,999)
        num2 = random.randint(0,99)
        num3 = random.randint(0,999999)
        num1 = str(num1).zfill(3)   
        num2 = str(num2).zfill(2)
        num3 = str(num3).zfill(6)   
        self.account_number = num1+'-'+num2+'-'+num3
        Account.count+=1
        
    def get_account_num(cls):
        print(cls.count)
        
    def deposit(self, money_in):
        if money_in >=1:
            self.balance+=money_in
            self.deposit_count+=1
            
            if self.deposit_count%5==0:
                self.balance *= 1.01
            
    def withdraw(self, money_out):
        if self.balance > money_out:
            self.balance-=money_out
            
    def display_info(self):
        print(f"은행이름 : {self.bank}")
        print(f"예금주 : {self.name}")
        print(f"계좌번호 : {self.account_number}")
        print(f"잔고 : {self.balance}원")
a=Account("A1",10000)
for i in range(12):
    a.deposit(10000)
    print(a.balance)

20000
30000
40000
50000
60600.0
70600.0
80600.0
90600.0
100600.0
111706.0
121706.0
131706.0


# 278 여러 객체 생성
Account 클래스로부터 3개 이상 인스턴스를 생성하고 생성된 인스턴스를 리스트에 저장해보세요.

In [27]:
class Account:
    count = 0
    def __init__(self, name, balance):   
        self.deposit_count = 0
        self.name = name
        self.balance = balance
        self.bank = "SC은행"
        num1 = random.randint(0,999)
        num2 = random.randint(0,99)
        num3 = random.randint(0,999999)
        num1 = str(num1).zfill(3)   
        num2 = str(num2).zfill(2)
        num3 = str(num3).zfill(6)   
        self.account_number = num1+'-'+num2+'-'+num3
        Account.count+=1
        
    def get_account_num(cls):
        print(cls.count)
        
    def deposit(self, money_in):
        if money_in >=1:
            self.balance+=money_in
            self.deposit_count+=1
            
            if self.deposit_count%5==0:
                self.balance *= 1.01
            
    def withdraw(self, money_out):
        if self.balance > money_out:
            self.balance-=money_out
            
    def display_info(self):
        print(f"은행이름 : {self.bank}")
        print(f"예금주 : {self.name}")
        print(f"계좌번호 : {self.account_number}")
        print(f"잔고 : {self.balance}원")

ans=[]
a1 = Account("A1",10000)
a2 = Account("A2",5000)
a3 = Account("A3",4000)

ans.append(a1)
ans.append(a2)
ans.append(a3)

print(ans)

[<__main__.Account object at 0x000002465700A0D0>, <__main__.Account object at 0x0000024657004070>, <__main__.Account object at 0x000002465700A100>]


# 279 객체 순회
반복문을 통해 리스트에 있는 객체를 순회하면서 잔고가 100만원 이상인 고객의 정보만 출력하세요.

In [28]:
class Account:
    count = 0
    def __init__(self, name, balance):   
        self.deposit_count = 0
        self.name = name
        self.balance = balance
        self.bank = "SC은행"
        num1 = random.randint(0,999)
        num2 = random.randint(0,99)
        num3 = random.randint(0,999999)
        num1 = str(num1).zfill(3)   
        num2 = str(num2).zfill(2)
        num3 = str(num3).zfill(6)   
        self.account_number = num1+'-'+num2+'-'+num3
        Account.count+=1
        
    def get_account_num(cls):
        print(cls.count)
        
    def deposit(self, money_in):
        if money_in >=1:
            self.balance+=money_in
            self.deposit_count+=1
            
            if self.deposit_count%5==0:
                self.balance *= 1.01
            
    def withdraw(self, money_out):
        if self.balance > money_out:
            self.balance-=money_out
            
    def display_info(self):
        print(f"은행이름 : {self.bank}")
        print(f"예금주 : {self.name}")
        print(f"계좌번호 : {self.account_number}")
        print(f"잔고 : {self.balance}원")

ans=[]
a1 = Account("A1",100000000000000)
a2 = Account("A2",5000)
a3 = Account("A3",4000)

ans.append(a1)
ans.append(a2)
ans.append(a3)

for i in ans:
    if i.balance >= 1000000:
        i.display_info()

은행이름 : SC은행
예금주 : A1
계좌번호 : 923-91-621318
잔고 : 100000000000000원


# 280 입출금 내역
입금과 출금 내역이 기록되도록 코드를 업데이트 하세요. 입금 내역과 출금 내역을 출력하는 deposit_history와 withdraw_history 메서드를 추가하세요.

In [29]:
import random


class Account:
    # class variable
    account_count = 0

    def __init__(self, name, balance):
        self.deposit_count = 0
        self.deposit_log = []
        self.withdraw_log = []

        self.name = name
        self.balance = balance
        self.bank = "SC은행"

        # 3-2-6
        num1 = random.randint(0, 999)
        num2 = random.randint(0, 99)
        num3 = random.randint(0, 999999)

        num1 = str(num1).zfill(3)  # 1 -> '1' -> '001'
        num2 = str(num2).zfill(2)  # 1 -> '1' -> '01'
        num3 = str(num3).zfill(6)  # 1 -> '1' -> '0000001'
        self.account_number = num1 + '-' + num2 + '-' + num3  # 001-01-000001
        Account.account_count += 1

    @classmethod
    def get_account_num(cls):
        print(cls.account_count)  # Account.account_count

    def deposit(self, amount):
        if amount >= 1:
            self.deposit_log.append(amount)
            self.balance += amount

            self.deposit_count += 1
            if self.deposit_count % 5 == 0:         # 5, 10, 15
                # 이자 지금
                self.balance = (self.balance * 1.01)


    def withdraw(self, amount):
        if self.balance > amount:
            self.withdraw_log.append(amount)
            self.balance -= amount

    def display_info(self):
        print("은행이름: ", self.bank)
        print("예금주: ", self.name)
        print("계좌번호: ", self.account_number)
        print("잔고: ", self.balance)

    def withdraw_history(self):
        for amount in self.withdraw_log:
            print(amount)

    def deposit_history(self):
        for amount in self.deposit_log:
            print(amount)


k = Account("Kim", 1000)
k.deposit(100)
k.deposit(200)
k.deposit(300)
k.deposit_history()

k.withdraw(100)
k.withdraw(200)
k.withdraw_history()


100
200
300
100
200


# 281 클래스 정의
다음 코드가 동작하도록 차 클래스를 정의하세요.

car = 차(2, 1000)

car.바퀴

2

car.가격

1000

In [33]:
class 차:
    def __init__(self,바퀴,가격):
        self.바퀴 = 바퀴
        self.가격 = 가격

car = 차(2, 1000)
print(car.바퀴)
print(car.가격)

2
1000


# 282 클래스 상속
차 클래스를 상속받은 자전차 클래스를 정의하세요.

In [34]:
class 차:
    def __init__(self,바퀴,가격):
        self.바퀴 = 바퀴
        self.가격 = 가격
        
class 자천차(차):
    pass

# 283 클래스 상속
다음 코드가 동작하도록 자전차 클래스를 정의하세요. 단 자전차 클래스는 차 클래스를 상속받습니다.

bicycle = 자전차(2, 100)

bicycle.가격

100

In [1]:
class 차:
    def __init__(self, 바퀴, 가격):
        self.바퀴 = 바퀴
        self.가격 = 가격


class 자전차(차):
    def __init__(self, 바퀴, 가격):
        self.바퀴 = 바퀴
        self.가격 = 가격
        
bicycle = 자전차(2, 100)
print(bicycle.가격)

100


# 284 클래스 상속
다음 코드가 동작하도록 자전차 클래스를 정의하세요. 단 자전차 클래스는 차 클래스를 상속받습니다.

bicycle = 자전차(2, 100, "시마노")

bicycle.구동계

시마노

In [2]:
class 차:
    def __init__(self, 바퀴, 가격):
        self.바퀴 = 바퀴
        self.가격 = 가격


class 자전차(차):
    def __init__(self, 바퀴, 가격,구동계):
        self.바퀴 = 바퀴
        self.가격 = 가격
        self.구동계 = 구동계
        
bicycle = 자전차(2, 100, "시마노")
print(bicycle.구동계)

시마노


# 285 클래스 상속
다음 코드가 동작하도록 차 클래스를 상속받는 자동차 클래스를 정의하세요.

car = 자동차(4, 1000)

car.정보()

바퀴수 4

가격 1000

In [7]:
class 차:
    def __init__(self, 바퀴, 가격):
        self.바퀴 = 바퀴
        self.가격 = 가격


class 자동차(차):
    def __init__(self, 바퀴, 가격):
        self.바퀴 = 바퀴
        self.가격 = 가격
    def 정보(self):
        print(f"바퀴수 {self.바퀴}")
        print(f"가격 {self.가격}")
car = 자동차(4, 1000)
car.정보()

바퀴수 4
가격 1000


# 286 부모 클래스 생성자 호출
다음 코드가 동작하도록 차 클래스를 수정하세요.

bicycle = 자전차(2, 100, "시마노")

bicycle.정보()

바퀴수 2

가격 100

In [8]:
class 차:
    def __init__(self, 바퀴, 가격):
        self.바퀴 = 바퀴
        self.가격 = 가격
    def 정보(self):
        print(f"바퀴수 {self.바퀴}")
        print(f"가격 {self.가격}")
        
class 자동차(차):
    def __init__(self,바퀴,가격):
        super().__init__(바퀴,가격)
        
        
class 자전차(차):
    def __init__(self, 바퀴, 가격,구동계):
        super().__init__(바퀴,가격)
        self.구동계 = 구동계
        
bicycle = 자전차(2,100,"시마노")
bicycle.정보()

바퀴수 2
가격 100


# 287 부모 클래스 메서드 호출
자전차의 정보() 메서드로 구동계 정보까지 출력하도록 수정해보세요.

bicycle = 자전차(2, 100, "시마노")

bicycle.정보()

바퀴수 2

가격 100

구동계 시마노

In [9]:
class 차:
    def __init__(self, 바퀴, 가격):
        self.바퀴 = 바퀴
        self.가격 = 가격
    def 정보(self):
        print(f"바퀴수 {self.바퀴}")
        print(f"가격 {self.가격}")
        
class 자동차(차):
    def __init__(self,바퀴,가격):
        super().__init__(바퀴,가격)
        
        
class 자전차(차):
    def __init__(self, 바퀴, 가격,구동계):
        super().__init__(바퀴,가격)
        self.구동계 = 구동계
    def 정보(self):
        super().정보()
        print(f"구동계 {self.구동계}")
        
bicycle = 자전차(2,100,"시마노")
bicycle.정보()

바퀴수 2
가격 100
구동계 시마노


# 288 오버라이딩
class 부모:
  
  def 호출(self):
    
    print("부모호출")

class 자식(부모):
  
  def 호출(self):
    
    print("자식호출")

나 = 자식()

나.호출()

답 : 자식호출

In [10]:
class 부모:
  def 호출(self):
    print("부모호출")

class 자식(부모):
  def 호출(self):
    print("자식호출")
나 = 자식()
나.호출()

자식호출


# 289 생성자
다음 코드의 실행 결과를 예상해보세요.

class 부모:
  
  def __init__(self):
    
    print("부모생성")

class 자식(부모):
  
  def __init__(self):
    
    print("자식생성")

답 : 자식생성

In [14]:
class 부모:
  def __init__(self):
    print("부모생성")

class 자식(부모):
  def __init__(self):
    print("자식생성")
    
나 = 자식()

자식생성


# 290 부모클래스 생성자 호출
다음 코드의 실행 결과를 예상해보세요.

class 부모:
  
  def __init__(self):
    
    print("부모생성")

class 자식(부모):
  
  def __init__(self):
    
    print("자식생성")
    
    super().__init__()

나 = 자식()


답 :

자식생성

부모생성

In [15]:
class 부모:
  def __init__(self):
    print("부모생성")

class 자식(부모):
  def __init__(self):
    print("자식생성")
    super().__init__()

나 = 자식()

자식생성
부모생성


# <12.파일 입출력과 예외처리>

# 291 파일쓰기
바탕화면에 '매수종목1.txt' 파일을 생성한 후 다음과 같이 종목코드를 파일에 써보세요.

005930

005380

035420

In [29]:
f = open("C:/Users\kbk/desktop\매수종목1.txt", mode="wt", encoding="utf-8")
f.write("005930\n")
f.write("005380\n")
f.write("035420")
f.close()

# 292 파일쓰기
바탕화면에 '매수종목2.txt' 파일을 생성한 후 다음과 같이 종목코드와 종목명을 파일에 써보세요.

005930 삼성전자

005380 현대차

035420 NAVER

In [30]:
f = open("C:/Users\kbk/desktop\매수종목2.txt", mode="wt", encoding="utf-8")
f.write("005930 삼성전자\n")
f.write("005380 현대차\n")
f.write("035420 NAVER")
f.close()

# 293 CSV 파일 쓰기
바탕화면에 '매수종목.csv' 파일을 생성한 후 다음과 같이 종목코드와 종목명을 파일에 써보세요. 인코딩은 'cp949'를 사용해야합니다.

In [31]:
import csv

f = open("C:/Users\kbk/desktop\매수종목.csv", mode="wt", encoding="cp949",
        newline='')
writer = csv.writer(f)
writer.writerow(["종목명","종목코드","PER"])
writer.writerow(["삼성전자","005930","15.59"])
writer.writerow(["NAVER","035420","55.82"])
f.close()

# 294 파일 읽기
바탕화면에 생성한 '매수종목1.txt' 파일을 읽은 후 종목코드를 리스트에 저장해보세요.

In [33]:
f = open("C:/Users\kbk/desktop\매수종목1.txt", encoding="utf-8")
lines = f.readlines()
codes=[]
for a in lines:
    code = a.strip()
    codes.append(code)
f.close()
    
print(codes)


['005930', '005380', '035420']


# 295 파일 읽기
바탕화면에 생성한 '매수종목2.txt' 파일을 읽은 후 종목코드와 종목명을 딕셔너리로 저장해보세요. 종목명을 key로 종목명을 value로 저장합니다.



In [40]:
f = open("C:/Users\kbk/desktop\매수종목2.txt", encoding="utf-8")
lines = f.readlines()
dic = {}
for line in lines:
    line = line.strip()
    k, v = line.split()
    dic[k] = v
f.close()

print(data)

{'005930': '삼성전자', '005380': '현대차', '035420': 'NAVER'}


# 296 예외처리
문자열 PER (Price to Earning Ratio) 값을 실수로 변환할 때 에러가 발생합니다. 예외처리를 통해 에러가 발생하는 PER은 0으로 출력하세요.

per = ["10.31", "", "8.00"]

for i in per:
    
    print(float(i))


In [41]:
per = ["10.31", "", "8.00"]
for i in per:
    try:
        print(float(i))
    except:print(0)

10.31
0
8.0


# 297 예외처리 및 리스트에 저장
문자열로 표현된 PER 값을 실수로 변환한 후 이를 새로운 리스트에 저장해보세요.

per = ["10.31", "", "8.00"]

for i in per:
    
    print(float(per))

In [42]:
per = ["10.31", "", "8.00"]
ans = []
for i in per:
    try:
        x=float(i)
    except:
        x=0
    ans.append(x)
print(ans)

[10.31, 0, 8.0]


# 298 특정 예외만 처리하기
어떤 값을 0으로 나누면 ZeroDivisionError 에러가 발생합니다. try ~ except로 모든 에러에 대해 예외처리하지 말고 ZeroDivisionError 에러만 예외처리해보세요.

In [43]:
try :
    b=3/0
except ZeroDivisionError:
    print("0으로 나눌수 없음")

0으로 나눌수 없음


# 299 예외의 메시지 출력하기
다음과 같은 코드 구조를 사용하면 예외 발생 시 에러 메시지를 변수로 바인딩할 수 있습니다.

try:
    
    실행코드

except 예외 as 변수:
    
    예외처리코드 

리스트의 인덱싱에 대해 에러를 출력해보세요.

data = [1, 2, 3]

for i in range(5)
    
    print(data[i])

In [45]:
data=[1,2,3]
for i in range(5):
    try:
        print(data[i])
    except IndexError :
        print("Index Error!!!")

1
2
3
Index Error!!!
Index Error!!!


# 300 try,except,else,finally 구조 사용해보기
파이썬 예외처리는 다음과 같은 구조를 가질 수 있습니다.

try:
    
    실행 코드

except:
    
    예외가 발생했을 때 수행할 코드

else:
    
    예외가 발생하지 않았을 때 수행할 코드

finally:
    
    예외 발생 여부와 상관없이 항상 수행할 코드

아래의 코드에 대해서 예외처리를 사용하고 try, except, else, finally에 적당한 코드
를 작성해봅시다. else와 finally는 적당한 문구를 print하시면 됩니다.

per = ["10.31", "", "8.00"]

for i in per:
    
    print(float(per))

In [47]:
per = ["10.31", "", "8.00"]
for i in per:
    try:
        print(float(i))
    except:
        print(0)
    else:
        print("예외가 발생하지 않음")
    finally:
        print("항상 마지막에 오는 코드\n")

10.31
예외가 발생하지 않음
항상 마지막에 오는 코드

0
항상 마지막에 오는 코드

8.0
예외가 발생하지 않음
항상 마지막에 오는 코드

